{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4151d2bcebe5c238",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T07:50:21.546396900Z",
     "start_time": "2024-01-23T07:50:21.546396900Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f83ae74642b3d5e",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psych_dfs = []\n",
    "other_dfs = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21d4ee7399a8c6fc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-15T08:20:32.780644400Z",
     "start_time": "2024-01-15T08:20:17.540341700Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "# fibromyalgia = pd.read_csv('Summary_statistics/raw/fibromyalgia.tsv', sep='\\t', comment='#') #need to map rsID\n",
    "# fatigue = pd.read_csv('Summary_statistics/raw/chronic_fatigue.tsv', sep='\\t', comment='#') #need to map rsID\n",
    "# gout = pd.read_csv('Summary_statistics/raw/gout.tsv', sep='\\t') #need to map rsID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6e87d7a674a67e6c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:37:01.921092700Z",
     "start_time": "2024-01-22T12:36:46.601890Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sjogrens_df= pd.read_csv('sjogrens_buildGRCh37.tsv', sep='\\t')\n",
    "#sjogrens_df = sjogrens_df[['variant_id', 'p_value']]\n",
    "sjogrens_df = sjogrens_df.rename(columns={'chromosome':'chromosome','variant_id':'variant_id','base_pair_location':'base_pair_location', 'other_allele':'other_allele', 'effect_allele':'effect_allele', 'effect_allele_frequency':'effect_allele_frequency', 'INFO':'INFO', 'beta':'beta', 'standard_error':'standard_error', 'T':'T', '-log10p':'-log10p', 'p_value':'p_value'})\n",
    "# other_dfs.append(sjogrens_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c7ce350e63a567ab",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:38:55.511717200Z",
     "start_time": "2024-01-22T12:37:45.724507100Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sjogrens_df.to_csv('sjogrens.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "fc54da49c8034dc2",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:37:05.784112800Z",
     "start_time": "2024-01-22T12:37:05.768309600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 9419702 entries, 0 to 9419701\n",
      "Data columns (total 12 columns):\n",
      " #   Column                   Dtype  \n",
      "---  ------                   -----  \n",
      " 0   chromosome               int64  \n",
      " 1   variant_id               object \n",
      " 2   base_pair_location       int64  \n",
      " 3   other_allele             object \n",
      " 4   effect_allele            object \n",
      " 5   effect_allele_frequency  float64\n",
      " 6   INFO                     float64\n",
      " 7   beta                     float64\n",
      " 8   standard_error           float64\n",
      " 9   T                        float64\n",
      " 10  -log10p                  float64\n",
      " 11  p_value                  float64\n",
      "dtypes: float64(7), int64(2), object(3)\n",
      "memory usage: 862.4+ MB\n"
     ]
    }
   ],
   "source": [
    "sjogrens_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a184843d98147bae",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>rs_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>130840091</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-1.321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8493</td>\n",
       "      <td>0.1866</td>\n",
       "      <td>rs2326918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7</td>\n",
       "      <td>145771806</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-1.294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8541</td>\n",
       "      <td>0.1956</td>\n",
       "      <td>rs6977693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>100009976</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8720</td>\n",
       "      <td>0.8532</td>\n",
       "      <td>rs12364336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>166367755</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>1.379</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7572</td>\n",
       "      <td>0.1680</td>\n",
       "      <td>rs12562373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>86737556</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1772</td>\n",
       "      <td>0.9208</td>\n",
       "      <td>rs2135099</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location effect_allele other_allele   beta  \\\n",
       "0           6           130840091             a            g -1.321   \n",
       "1           7           145771806             t            c -1.294   \n",
       "2          11           100009976             a            g  0.185   \n",
       "3           1           166367755             a            g  1.379   \n",
       "4          14            86737556             a            g  0.099   \n",
       "\n",
       "   standard_error  effect_allele_frequency  p_value       rs_id  \n",
       "0             NaN                   0.8493   0.1866   rs2326918  \n",
       "1             NaN                   0.8541   0.1956   rs6977693  \n",
       "2             NaN                   0.8720   0.8532  rs12364336  \n",
       "3             NaN                   0.7572   0.1680  rs12562373  \n",
       "4             NaN                   0.1772   0.9208   rs2135099  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "epilepsy_df = pd.read_csv('epilepsy.tsv', sep='\\t', nrows=5)\n",
    "#epilepsy_df = epilepsy_df[['rs_id', 'p_value']]\n",
    "#other_dfs.append(epilepsy_df)\n",
    "epilepsy_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1b65773c5af44995",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1lele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs12562034</td>\n",
       "      <td>768448</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.915</td>\n",
       "      <td>0.3604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7518545</td>\n",
       "      <td>769963</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.904</td>\n",
       "      <td>0.3660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs12124819</td>\n",
       "      <td>776546</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.075</td>\n",
       "      <td>0.9405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs117086422</td>\n",
       "      <td>845635</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-0.777</td>\n",
       "      <td>0.4374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs28612348</td>\n",
       "      <td>846078</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-0.826</td>\n",
       "      <td>0.4090</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR          SNP      BP A1lele1 Allele2  Zscore  P-value\n",
       "0    1   rs12562034  768448       a       g  -0.915   0.3604\n",
       "1    1    rs7518545  769963       a       g  -0.904   0.3660\n",
       "2    1   rs12124819  776546       a       g  -0.075   0.9405\n",
       "3    1  rs117086422  845635       t       c  -0.777   0.4374\n",
       "4    1   rs28612348  846078       t       c  -0.826   0.4090"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lupus_df = pd.read_csv('systemic_lupus.txt', sep='\\t', nrows=5)\n",
    "\"\"\" lupus_df = lupus_df[['SNP', 'P-value']]\n",
    "lupus_df = lupus_df.rename(columns={'SNP':'rs_id','P-value':'p_value'})\n",
    "other_dfs.append(lupus_df) \"\"\"\n",
    "lupus_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a5a356aa146bc32a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>p_value</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs6689308</td>\n",
       "      <td>1</td>\n",
       "      <td>1029805</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.510882</td>\n",
       "      <td>-0.019390</td>\n",
       "      <td>0.029493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs6687776</td>\n",
       "      <td>1</td>\n",
       "      <td>1030565</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.541961</td>\n",
       "      <td>-0.017982</td>\n",
       "      <td>0.029485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs6678318</td>\n",
       "      <td>1</td>\n",
       "      <td>1030633</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.541961</td>\n",
       "      <td>-0.017982</td>\n",
       "      <td>0.029485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs9651270</td>\n",
       "      <td>1</td>\n",
       "      <td>1031973</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.485029</td>\n",
       "      <td>-0.020591</td>\n",
       "      <td>0.029490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs6699750</td>\n",
       "      <td>1</td>\n",
       "      <td>1032954</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.489730</td>\n",
       "      <td>-0.020370</td>\n",
       "      <td>0.029490</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0  rs6689308           1             1029805             G            A   \n",
       "1  rs6687776           1             1030565             T            C   \n",
       "2  rs6678318           1             1030633             A            G   \n",
       "3  rs9651270           1             1031973             T            C   \n",
       "4  rs6699750           1             1032954             G            A   \n",
       "\n",
       "    p_value      beta  standard_error  \n",
       "0  0.510882 -0.019390        0.029493  \n",
       "1  0.541961 -0.017982        0.029485  \n",
       "2  0.541961 -0.017982        0.029485  \n",
       "3  0.485029 -0.020591        0.029490  \n",
       "4  0.489730 -0.020370        0.029490  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prim_bil_df = pd.read_csv('biliary_chirrosis.tsv', sep='\\t', comment='#', nrows=5)\n",
    "\"\"\" prim_bil_df = prim_bil_df[['variant_id', 'p_value']]\n",
    "prim_bil_df = prim_bil_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(prim_bil_df) \"\"\"\n",
    "prim_bil_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2f6ca9eff20f88c1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-17T09:28:00.232235Z",
     "start_time": "2024-01-17T09:27:37.096483600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs533090414</td>\n",
       "      <td>1</td>\n",
       "      <td>18849</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.026465</td>\n",
       "      <td>0.536412</td>\n",
       "      <td>2.606450</td>\n",
       "      <td>0.110</td>\n",
       "      <td>-0.003283</td>\n",
       "      <td>0.002005</td>\n",
       "      <td>2.681520</td>\n",
       "      <td>0.100</td>\n",
       "      <td>2.685780</td>\n",
       "      <td>0.100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs568927205</td>\n",
       "      <td>1</td>\n",
       "      <td>54712</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTTC</td>\n",
       "      <td>0.415825</td>\n",
       "      <td>0.538572</td>\n",
       "      <td>0.706876</td>\n",
       "      <td>0.400</td>\n",
       "      <td>0.000614</td>\n",
       "      <td>0.000646</td>\n",
       "      <td>0.903599</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.807515</td>\n",
       "      <td>0.370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>82133</td>\n",
       "      <td>0</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>0.020380</td>\n",
       "      <td>0.506777</td>\n",
       "      <td>3.242910</td>\n",
       "      <td>0.072</td>\n",
       "      <td>-0.004445</td>\n",
       "      <td>0.002356</td>\n",
       "      <td>3.560040</td>\n",
       "      <td>0.059</td>\n",
       "      <td>3.071210</td>\n",
       "      <td>0.080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs28619159</td>\n",
       "      <td>1</td>\n",
       "      <td>91421</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.981122</td>\n",
       "      <td>0.521572</td>\n",
       "      <td>0.000772</td>\n",
       "      <td>0.980</td>\n",
       "      <td>-0.000627</td>\n",
       "      <td>0.002397</td>\n",
       "      <td>0.068440</td>\n",
       "      <td>0.790</td>\n",
       "      <td>0.034140</td>\n",
       "      <td>0.850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs375085441</td>\n",
       "      <td>1</td>\n",
       "      <td>91551</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>0.989647</td>\n",
       "      <td>0.548005</td>\n",
       "      <td>5.591080</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.007733</td>\n",
       "      <td>0.003366</td>\n",
       "      <td>5.276900</td>\n",
       "      <td>0.022</td>\n",
       "      <td>5.337760</td>\n",
       "      <td>0.021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location  GENPOS effect_allele  \\\n",
       "0  rs533090414           1               18849       0             C   \n",
       "1  rs568927205           1               54712       0             T   \n",
       "2          NaN           1               82133       0           CAA   \n",
       "3   rs28619159           1               91421       0             T   \n",
       "4  rs375085441           1               91551       0             G   \n",
       "\n",
       "  other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "0            G                 0.026465  0.536412      2.606450     0.110   \n",
       "1        TTTTC                 0.415825  0.538572      0.706876     0.400   \n",
       "2            C                 0.020380  0.506777      3.242910     0.072   \n",
       "3            C                 0.981122  0.521572      0.000772     0.980   \n",
       "4           GT                 0.989647  0.548005      5.591080     0.018   \n",
       "\n",
       "       beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "0 -0.003283        0.002005            2.681520           0.100   \n",
       "1  0.000614        0.000646            0.903599           0.340   \n",
       "2 -0.004445        0.002356            3.560040           0.059   \n",
       "3 -0.000627        0.002397            0.068440           0.790   \n",
       "4  0.007733        0.003366            5.276900           0.022   \n",
       "\n",
       "   CHISQ_BOLT_LMM  p_value  \n",
       "0        2.685780    0.100  \n",
       "1        0.807515    0.370  \n",
       "2        3.071210    0.080  \n",
       "3        0.034140    0.850  \n",
       "4        5.337760    0.021  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thyroid_df = pd.read_csv('thyroid_buildGRCh37.tsv', sep='\\t', comment='#', nrows=5)\n",
    "# thyroid_df = thyroid_df[['variant_id', 'p_value']]\n",
    "# thyroid_df = thyroid_df.rename(columns={'variant_id':'rs_id'})\n",
    "# other_dfs.append(thyroid_df)\n",
    "thyroid_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f008872c9ee88173",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T13:01:48.451685300Z",
     "start_time": "2024-01-22T13:01:47.169691800Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scleroderma_df = pd.read_csv('systemic_scleroderma.txt', header=0, delim_whitespace=True)\n",
    "scleroderma_df = scleroderma_df[['SNP', 'P', 'A1', 'A2', 'OR']]\n",
    "# scleroderma_df = scleroderma_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "# other_dfs.append(scleroderma_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a544155ade05b6f6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>effect_allele_frequency_cases</th>\n",
       "      <th>effect_allele_frequency_controls</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs61769340</td>\n",
       "      <td>1</td>\n",
       "      <td>727717</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.132900</td>\n",
       "      <td>0.144688</td>\n",
       "      <td>0.132293</td>\n",
       "      <td>0.1063</td>\n",
       "      <td>0.0522</td>\n",
       "      <td>0.041610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>rs116030099</td>\n",
       "      <td>1</td>\n",
       "      <td>787290</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.079988</td>\n",
       "      <td>0.063801</td>\n",
       "      <td>0.080822</td>\n",
       "      <td>-0.1757</td>\n",
       "      <td>0.0736</td>\n",
       "      <td>0.017020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>rs148120343</td>\n",
       "      <td>1</td>\n",
       "      <td>794707</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.044400</td>\n",
       "      <td>0.036839</td>\n",
       "      <td>0.044790</td>\n",
       "      <td>0.1894</td>\n",
       "      <td>0.0966</td>\n",
       "      <td>0.049970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>rs79010578</td>\n",
       "      <td>1</td>\n",
       "      <td>800909</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.095547</td>\n",
       "      <td>0.084357</td>\n",
       "      <td>0.096123</td>\n",
       "      <td>0.1344</td>\n",
       "      <td>0.0653</td>\n",
       "      <td>0.039550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>rs61768170</td>\n",
       "      <td>1</td>\n",
       "      <td>821054</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.105143</td>\n",
       "      <td>0.102776</td>\n",
       "      <td>0.105265</td>\n",
       "      <td>0.1569</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.009184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "9     rs61769340           1              727717             G            C   \n",
       "52   rs116030099           1              787290             C            T   \n",
       "59   rs148120343           1              794707             C            T   \n",
       "76    rs79010578           1              800909             A            T   \n",
       "117   rs61768170           1              821054             C            G   \n",
       "\n",
       "     effect_allele_frequency  effect_allele_frequency_cases  \\\n",
       "9                   0.132900                       0.144688   \n",
       "52                  0.079988                       0.063801   \n",
       "59                  0.044400                       0.036839   \n",
       "76                  0.095547                       0.084357   \n",
       "117                 0.105143                       0.102776   \n",
       "\n",
       "     effect_allele_frequency_controls    beta  standard_error   p_value  \n",
       "9                            0.132293  0.1063          0.0522  0.041610  \n",
       "52                           0.080822 -0.1757          0.0736  0.017020  \n",
       "59                           0.044790  0.1894          0.0966  0.049970  \n",
       "76                           0.096123  0.1344          0.0653  0.039550  \n",
       "117                          0.105265  0.1569          0.0602  0.009184  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myasthenia_df = pd.read_csv('myastheria_gravis.tsv', sep='\\t', comment='#')\n",
    "myasthenia_df = myasthenia_df[myasthenia_df['p_value'] <= 0.05]\n",
    "\"\"\" myasthenia_df = myasthenia_df[['variant_id', 'p_value']]\n",
    "myasthenia_df = myasthenia_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(myasthenia_df) \"\"\"\n",
    "myasthenia_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9c830a6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "myasthenia_df.to_csv('myasthenia_gravis_filtered.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ebde0194b6646a7a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id_hg19</th>\n",
       "      <th>base_pair_location_grch38</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs144804129</td>\n",
       "      <td>10</td>\n",
       "      <td>100000122</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.2710</td>\n",
       "      <td>0.4979</td>\n",
       "      <td>0.5863</td>\n",
       "      <td>10_100000122_T_A</td>\n",
       "      <td>98240365.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs6602381</td>\n",
       "      <td>10</td>\n",
       "      <td>10000018</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0008</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>0.9550</td>\n",
       "      <td>10_10000018_A_G</td>\n",
       "      <td>9958055.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11442554</td>\n",
       "      <td>10</td>\n",
       "      <td>100000554</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.1945</td>\n",
       "      <td>10_100000554_A_AT</td>\n",
       "      <td>98240797.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs112832083</td>\n",
       "      <td>10</td>\n",
       "      <td>100000588</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.5309</td>\n",
       "      <td>0.9955</td>\n",
       "      <td>0.5938</td>\n",
       "      <td>10_100000588_T_C</td>\n",
       "      <td>98240831.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7899632</td>\n",
       "      <td>10</td>\n",
       "      <td>100000625</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0070</td>\n",
       "      <td>0.0153</td>\n",
       "      <td>0.6492</td>\n",
       "      <td>10_100000625_A_G</td>\n",
       "      <td>98240868.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0  rs144804129          10           100000122             A            T   \n",
       "1    rs6602381          10            10000018             G            A   \n",
       "2   rs11442554          10           100000554            AT            A   \n",
       "3  rs112832083          10           100000588             C            T   \n",
       "4    rs7899632          10           100000625             G            A   \n",
       "\n",
       "     beta  standard_error  p_value    variant_id_hg19  \\\n",
       "0 -0.2710          0.4979   0.5863   10_100000122_T_A   \n",
       "1 -0.0008          0.0143   0.9550    10_10000018_A_G   \n",
       "2  0.0281          0.0217   0.1945  10_100000554_A_AT   \n",
       "3 -0.5309          0.9955   0.5938   10_100000588_T_C   \n",
       "4 -0.0070          0.0153   0.6492   10_100000625_A_G   \n",
       "\n",
       "   base_pair_location_grch38  \n",
       "0                 98240365.0  \n",
       "1                  9958055.0  \n",
       "2                 98240797.0  \n",
       "3                 98240831.0  \n",
       "4                 98240868.0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rh_arthritis_df = pd.read_csv('rheumatoid_arthritis.tsv', sep='\\t', comment='#')\n",
    "\"\"\" rh_arthritis_df = rh_arthritis_df[['variant_id', 'p_value']]\n",
    "rh_arthritis_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(rh_arthritis_df) \"\"\"\n",
    "rh_arthritis_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f5483e0a41f87001",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs9331226</td>\n",
       "      <td>1</td>\n",
       "      <td>992819</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.036136</td>\n",
       "      <td>0.025892</td>\n",
       "      <td>0.162817</td>\n",
       "      <td>0.422199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs13303240</td>\n",
       "      <td>1</td>\n",
       "      <td>993360</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.035804</td>\n",
       "      <td>0.025815</td>\n",
       "      <td>0.165457</td>\n",
       "      <td>0.428449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs9442393</td>\n",
       "      <td>1</td>\n",
       "      <td>995481</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.043564</td>\n",
       "      <td>0.025660</td>\n",
       "      <td>0.089555</td>\n",
       "      <td>0.437241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs28479311</td>\n",
       "      <td>1</td>\n",
       "      <td>996248</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.028753</td>\n",
       "      <td>0.035137</td>\n",
       "      <td>0.413193</td>\n",
       "      <td>0.157065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs113991671</td>\n",
       "      <td>1</td>\n",
       "      <td>997769</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>-0.041790</td>\n",
       "      <td>0.028246</td>\n",
       "      <td>0.139005</td>\n",
       "      <td>0.271902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0    rs9331226           1              992819             G            A   \n",
       "1   rs13303240           1              993360             C            A   \n",
       "2    rs9442393           1              995481             T            G   \n",
       "3   rs28479311           1              996248             T            C   \n",
       "4  rs113991671           1              997769             C           CG   \n",
       "\n",
       "       beta  standard_error   p_value  effect_allele_frequency  \n",
       "0 -0.036136        0.025892  0.162817                 0.422199  \n",
       "1 -0.035804        0.025815  0.165457                 0.428449  \n",
       "2 -0.043564        0.025660  0.089555                 0.437241  \n",
       "3 -0.028753        0.035137  0.413193                 0.157065  \n",
       "4 -0.041790        0.028246  0.139005                 0.271902  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ms_df= pd.read_csv('ms.tsv', sep='\\t', comment='#', nrows=5)\n",
    "\"\"\" ms_df = ms_df[['variant_id', 'p_value']]\n",
    "ms_df = ms_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(ms_df) \"\"\"\n",
    "ms_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "34d8e130067df282",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T07:51:15.689525200Z",
     "start_time": "2024-01-23T07:51:12.007797300Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>SNP_HRC_ID</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>ci_lower</th>\n",
       "      <th>ci_upper</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>762320</td>\n",
       "      <td>1:762320:C:T</td>\n",
       "      <td>rs75333668</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.15582</td>\n",
       "      <td>0.341344</td>\n",
       "      <td>0.592022</td>\n",
       "      <td>2.25653</td>\n",
       "      <td>0.424232</td>\n",
       "      <td>0.671396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>765028</td>\n",
       "      <td>1:765028:C:T</td>\n",
       "      <td>rs2519025</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.28589</td>\n",
       "      <td>0.418656</td>\n",
       "      <td>0.566036</td>\n",
       "      <td>2.92123</td>\n",
       "      <td>0.600618</td>\n",
       "      <td>0.548094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>780229</td>\n",
       "      <td>1:780229:T:C</td>\n",
       "      <td>rs12565032</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.30387</td>\n",
       "      <td>0.419182</td>\n",
       "      <td>0.573356</td>\n",
       "      <td>2.96511</td>\n",
       "      <td>0.632978</td>\n",
       "      <td>0.526748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>782677</td>\n",
       "      <td>1:782677:G:T</td>\n",
       "      <td>rs140041012</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.21011</td>\n",
       "      <td>0.407450</td>\n",
       "      <td>0.544505</td>\n",
       "      <td>2.68934</td>\n",
       "      <td>0.468054</td>\n",
       "      <td>0.639746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865628</td>\n",
       "      <td>1:865628:G:A</td>\n",
       "      <td>rs41285790</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.15970</td>\n",
       "      <td>0.165675</td>\n",
       "      <td>0.838150</td>\n",
       "      <td>1.60460</td>\n",
       "      <td>0.894276</td>\n",
       "      <td>0.371174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location    SNP_HRC_ID   variant_id REF ALT  \\\n",
       "0           1              762320  1:762320:C:T   rs75333668   C   T   \n",
       "1           1              765028  1:765028:C:T    rs2519025   C   T   \n",
       "2           1              780229  1:780229:T:C   rs12565032   T   C   \n",
       "3           1              782677  1:782677:G:T  rs140041012   G   T   \n",
       "4           1              865628  1:865628:G:A   rs41285790   G   A   \n",
       "\n",
       "  effect_allele TEST  OBS_CT  odds_ratio  LOG(OR)_SE  ci_lower  ci_upper  \\\n",
       "0             T  ADD   10533     1.15582    0.341344  0.592022   2.25653   \n",
       "1             T  ADD   10533     1.28589    0.418656  0.566036   2.92123   \n",
       "2             C  ADD   10533     1.30387    0.419182  0.573356   2.96511   \n",
       "3             T  ADD   10533     1.21011    0.407450  0.544505   2.68934   \n",
       "4             A  ADD   10533     1.15970    0.165675  0.838150   1.60460   \n",
       "\n",
       "     Z_STAT   p_value  \n",
       "0  0.424232  0.671396  \n",
       "1  0.600618  0.548094  \n",
       "2  0.632978  0.526748  \n",
       "3  0.468054  0.639746  \n",
       "4  0.894276  0.371174  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parkinsons_df = pd.read_csv('parkinson.tsv',sep='\\t', comment='#', nrows=5)\n",
    "# parkinsons_df = parkinsons_df[['variant_id', 'p_value']]\n",
    "# parkinsons_df.rename(columns={'variant_id':'rs_id'})\n",
    "# other_dfs.append(parkinsons_df)\n",
    "parkinsons_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b1611444a91dc1c8",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs533090414</td>\n",
       "      <td>1</td>\n",
       "      <td>18849</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.026465</td>\n",
       "      <td>0.536412</td>\n",
       "      <td>1.540600</td>\n",
       "      <td>0.210</td>\n",
       "      <td>-0.001750</td>\n",
       "      <td>0.001460</td>\n",
       "      <td>1.436110</td>\n",
       "      <td>0.230</td>\n",
       "      <td>1.469600</td>\n",
       "      <td>0.230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs568927205</td>\n",
       "      <td>1</td>\n",
       "      <td>54712</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTTC</td>\n",
       "      <td>0.415825</td>\n",
       "      <td>0.538572</td>\n",
       "      <td>0.237088</td>\n",
       "      <td>0.630</td>\n",
       "      <td>-0.000248</td>\n",
       "      <td>0.000470</td>\n",
       "      <td>0.277492</td>\n",
       "      <td>0.600</td>\n",
       "      <td>0.298682</td>\n",
       "      <td>0.580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>82133</td>\n",
       "      <td>0</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>0.020380</td>\n",
       "      <td>0.506777</td>\n",
       "      <td>0.001665</td>\n",
       "      <td>0.970</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.001715</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.000376</td>\n",
       "      <td>0.980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs28619159</td>\n",
       "      <td>1</td>\n",
       "      <td>91421</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.981122</td>\n",
       "      <td>0.521572</td>\n",
       "      <td>2.825450</td>\n",
       "      <td>0.093</td>\n",
       "      <td>0.002980</td>\n",
       "      <td>0.001746</td>\n",
       "      <td>2.913510</td>\n",
       "      <td>0.088</td>\n",
       "      <td>3.045260</td>\n",
       "      <td>0.081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs375085441</td>\n",
       "      <td>1</td>\n",
       "      <td>91551</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>0.989647</td>\n",
       "      <td>0.548005</td>\n",
       "      <td>0.554736</td>\n",
       "      <td>0.460</td>\n",
       "      <td>0.001930</td>\n",
       "      <td>0.002451</td>\n",
       "      <td>0.619945</td>\n",
       "      <td>0.430</td>\n",
       "      <td>0.602850</td>\n",
       "      <td>0.440</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location  GENPOS effect_allele  \\\n",
       "0  rs533090414           1               18849       0             C   \n",
       "1  rs568927205           1               54712       0             T   \n",
       "2          NaN           1               82133       0           CAA   \n",
       "3   rs28619159           1               91421       0             T   \n",
       "4  rs375085441           1               91551       0             G   \n",
       "\n",
       "  other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "0            G                 0.026465  0.536412      1.540600     0.210   \n",
       "1        TTTTC                 0.415825  0.538572      0.237088     0.630   \n",
       "2            C                 0.020380  0.506777      0.001665     0.970   \n",
       "3            C                 0.981122  0.521572      2.825450     0.093   \n",
       "4           GT                 0.989647  0.548005      0.554736     0.460   \n",
       "\n",
       "       beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "0 -0.001750        0.001460            1.436110           0.230   \n",
       "1 -0.000248        0.000470            0.277492           0.600   \n",
       "2  0.000006        0.001715            0.000012           1.000   \n",
       "3  0.002980        0.001746            2.913510           0.088   \n",
       "4  0.001930        0.002451            0.619945           0.430   \n",
       "\n",
       "   CHISQ_BOLT_LMM  p_value  \n",
       "0        1.469600    0.230  \n",
       "1        0.298682    0.580  \n",
       "2        0.000376    0.980  \n",
       "3        3.045260    0.081  \n",
       "4        0.602850    0.440  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "migraine_df = pd.read_csv('migraine.tsv',sep='\\t', comment='#', nrows=5)\n",
    "\"\"\" migraine_df = migraine_df[['variant_id', 'p_value']]\n",
    "migraine_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(migraine_df) \"\"\"\n",
    "migraine_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4d271fe5b78d60a7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T09:42:03.767827600Z",
     "start_time": "2024-01-16T09:41:57.646604100Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>134897743</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.1725</td>\n",
       "      <td>-0.1574</td>\n",
       "      <td>0.1952</td>\n",
       "      <td>0.4199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>8600861</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.4160</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0.1511</td>\n",
       "      <td>0.3904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>100009976</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8734</td>\n",
       "      <td>0.3009</td>\n",
       "      <td>0.2185</td>\n",
       "      <td>0.1685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>3840048</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5911</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.1487</td>\n",
       "      <td>0.9941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>20776538</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.2802</td>\n",
       "      <td>-0.0839</td>\n",
       "      <td>0.1670</td>\n",
       "      <td>0.6155</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0          11           134897743             c            g   \n",
       "1          16             8600861             a            c   \n",
       "2          11           100009976             a            g   \n",
       "3          12             3840048             a            g   \n",
       "4           7            20776538             a            c   \n",
       "\n",
       "   effect_allele_frequency    beta  standard_error  p_value  \n",
       "0                   0.1725 -0.1574          0.1952   0.4199  \n",
       "1                   0.4160  0.1298          0.1511   0.3904  \n",
       "2                   0.8734  0.3009          0.2185   0.1685  \n",
       "3                   0.5911  0.0011          0.1487   0.9941  \n",
       "4                   0.2802 -0.0839          0.1670   0.6155  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fibromyalgia_df = pd.read_csv('fibromyalgia_buildGRCh37.tsv', sep='\\t')\n",
    "fibromyalgia_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8a6184f81415d85b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T10:47:39.964581400Z",
     "start_time": "2024-01-16T10:47:18.407440500Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs189259765</td>\n",
       "      <td>1</td>\n",
       "      <td>536095</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.723683</td>\n",
       "      <td>0.508780</td>\n",
       "      <td>4.99315</td>\n",
       "      <td>0.0250</td>\n",
       "      <td>0.000475</td>\n",
       "      <td>0.000212</td>\n",
       "      <td>5.01303</td>\n",
       "      <td>0.0250</td>\n",
       "      <td>5.03554</td>\n",
       "      <td>0.0250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs200129875</td>\n",
       "      <td>1</td>\n",
       "      <td>756257</td>\n",
       "      <td>0.004906</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.592892</td>\n",
       "      <td>0.525374</td>\n",
       "      <td>6.90706</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>0.000188</td>\n",
       "      <td>6.78899</td>\n",
       "      <td>0.0092</td>\n",
       "      <td>6.75710</td>\n",
       "      <td>0.0093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs57181708</td>\n",
       "      <td>1</td>\n",
       "      <td>809876</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.895195</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.43869</td>\n",
       "      <td>0.0350</td>\n",
       "      <td>0.000461</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>4.42652</td>\n",
       "      <td>0.0350</td>\n",
       "      <td>4.48090</td>\n",
       "      <td>0.0340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs72888853</td>\n",
       "      <td>1</td>\n",
       "      <td>815421</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.984519</td>\n",
       "      <td>0.558540</td>\n",
       "      <td>7.15932</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>-0.001951</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>7.21962</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>7.21795</td>\n",
       "      <td>0.0072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs535862693</td>\n",
       "      <td>1</td>\n",
       "      <td>815422</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.984519</td>\n",
       "      <td>0.558540</td>\n",
       "      <td>7.15932</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>-0.001951</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>7.21962</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>7.21795</td>\n",
       "      <td>0.0072</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location    GENPOS effect_allele  \\\n",
       "0  rs189259765           1              536095  0.000000             C   \n",
       "1  rs200129875           1              756257  0.004906             A   \n",
       "2   rs57181708           1              809876  0.006208             A   \n",
       "3   rs72888853           1              815421  0.006208             T   \n",
       "4  rs535862693           1              815422  0.006208             A   \n",
       "\n",
       "  other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "0            T                 0.723683  0.508780       4.99315    0.0250   \n",
       "1            C                 0.592892  0.525374       6.90706    0.0086   \n",
       "2            G                 0.895195  1.000000       4.43869    0.0350   \n",
       "3            C                 0.984519  0.558540       7.15932    0.0075   \n",
       "4            G                 0.984519  0.558540       7.15932    0.0075   \n",
       "\n",
       "       beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "0  0.000475        0.000212             5.01303          0.0250   \n",
       "1  0.000491        0.000188             6.78899          0.0092   \n",
       "2  0.000461        0.000219             4.42652          0.0350   \n",
       "3 -0.001951        0.000726             7.21962          0.0072   \n",
       "4 -0.001951        0.000726             7.21962          0.0072   \n",
       "\n",
       "   CHISQ_BOLT_LMM  p_value  \n",
       "0         5.03554   0.0250  \n",
       "1         6.75710   0.0093  \n",
       "2         4.48090   0.0340  \n",
       "3         7.21795   0.0072  \n",
       "4         7.21795   0.0072  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fatigue_df = pd.read_csv('fatigue_buildGRCh37.tsv', sep='\\t')\n",
    "fatigue_df = fatigue_df.loc[fatigue_df['p_value'] <= 0.05]\n",
    "fatigue_df.reset_index(drop=True, inplace=True)\n",
    "\"\"\" fatigue_df = fatigue_df[['variant_id', 'p_value']]\n",
    "fatigue_df = fatigue_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(fatigue_df)\n",
    " \"\"\"\n",
    "fatigue_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d365b3e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "fatigue_df.to_csv('fatigue_filtered_by_pval.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ed39d47fdf783ba",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ibs_df = pd.read_csv('Summary_statistics/raw/IBS.tsv', sep='\\t', comment='#')\n",
    "ibs_df = ibs_df[['variant_id', 'p_value']]\n",
    "ibs_df = ibs_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(ibs_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e530c6b2ca0853eb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-15T11:06:23.465586100Z",
     "start_time": "2024-01-15T11:06:02.056037600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs533316052</td>\n",
       "      <td>1</td>\n",
       "      <td>834430</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.981954</td>\n",
       "      <td>0.902227</td>\n",
       "      <td>4.88949</td>\n",
       "      <td>0.0270</td>\n",
       "      <td>-0.002142</td>\n",
       "      <td>0.000940</td>\n",
       "      <td>5.19438</td>\n",
       "      <td>0.0230</td>\n",
       "      <td>4.99975</td>\n",
       "      <td>0.0250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs565071104</td>\n",
       "      <td>1</td>\n",
       "      <td>845509</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.840534</td>\n",
       "      <td>0.738628</td>\n",
       "      <td>5.25631</td>\n",
       "      <td>0.0220</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>5.19605</td>\n",
       "      <td>0.0230</td>\n",
       "      <td>5.75915</td>\n",
       "      <td>0.0160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs145881258</td>\n",
       "      <td>1</td>\n",
       "      <td>876067</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.928461</td>\n",
       "      <td>0.844886</td>\n",
       "      <td>6.86200</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>0.001325</td>\n",
       "      <td>0.000503</td>\n",
       "      <td>6.94770</td>\n",
       "      <td>0.0084</td>\n",
       "      <td>6.74929</td>\n",
       "      <td>0.0094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>885882</td>\n",
       "      <td>0.006287</td>\n",
       "      <td>GAC</td>\n",
       "      <td>G</td>\n",
       "      <td>0.974464</td>\n",
       "      <td>0.887075</td>\n",
       "      <td>6.18209</td>\n",
       "      <td>0.0130</td>\n",
       "      <td>0.001964</td>\n",
       "      <td>0.000801</td>\n",
       "      <td>6.00773</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>5.85623</td>\n",
       "      <td>0.0160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs148363371</td>\n",
       "      <td>1</td>\n",
       "      <td>889876</td>\n",
       "      <td>0.006370</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.989011</td>\n",
       "      <td>0.955070</td>\n",
       "      <td>3.69856</td>\n",
       "      <td>0.0540</td>\n",
       "      <td>0.002260</td>\n",
       "      <td>0.001166</td>\n",
       "      <td>3.75791</td>\n",
       "      <td>0.0530</td>\n",
       "      <td>3.84921</td>\n",
       "      <td>0.0500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location    GENPOS effect_allele  \\\n",
       "0  rs533316052           1              834430  0.006208             G   \n",
       "1  rs565071104           1              845509  0.006208             T   \n",
       "2  rs145881258           1              876067  0.006208             C   \n",
       "3          NaN           1              885882  0.006287           GAC   \n",
       "4  rs148363371           1              889876  0.006370             C   \n",
       "\n",
       "  other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "0            A                 0.981954  0.902227       4.88949    0.0270   \n",
       "1            G                 0.840534  0.738628       5.25631    0.0220   \n",
       "2            T                 0.928461  0.844886       6.86200    0.0088   \n",
       "3            G                 0.974464  0.887075       6.18209    0.0130   \n",
       "4            G                 0.989011  0.955070       3.69856    0.0540   \n",
       "\n",
       "       beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "0 -0.002142        0.000940             5.19438          0.0230   \n",
       "1  0.000861        0.000378             5.19605          0.0230   \n",
       "2  0.001325        0.000503             6.94770          0.0084   \n",
       "3  0.001964        0.000801             6.00773          0.0140   \n",
       "4  0.002260        0.001166             3.75791          0.0530   \n",
       "\n",
       "   CHISQ_BOLT_LMM  p_value  \n",
       "0         4.99975   0.0250  \n",
       "1         5.75915   0.0160  \n",
       "2         6.74929   0.0094  \n",
       "3         5.85623   0.0160  \n",
       "4         3.84921   0.0500  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gout_df = pd.read_csv('gout_buildGRCh37.tsv', sep='\\t')\n",
    "gout_df = gout_df.loc[gout_df['p_value'] <= 0.05]\n",
    "gout_df.reset_index(inplace=True, drop= True)\n",
    "gout_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f7124516",
   "metadata": {},
   "outputs": [],
   "source": [
    "gout_df.to_csv('gout_filtered_by_pval.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e89bc4c6b43060a6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Psychiatric disorder GWAS dataset\n",
    "\n",
    "# alz = pd.read_csv('Summary_statistics/raw/PGCALZ2sumstatsExcluding23andMe.txt', sep='\\t') #need to map rsID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e04651a8e7dd7e18",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "adhd_df = pd.read_csv('Summary_statistics/raw/ADHD_meta_Jan2022_iPSYCH1_iPSYCH2_deCODE_PGC.meta', sep=' ')\n",
    "adhd_df = adhd_df[['SNP', 'P']]\n",
    "adhd_df = adhd_df.rename(columns={'SNP': 'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(adhd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b1105d2174a37f15",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T14:41:24.956773200Z",
     "start_time": "2024-01-16T14:41:16.865298100Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNPID</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Freq1</th>\n",
       "      <th>Effect</th>\n",
       "      <th>StdErr</th>\n",
       "      <th>P.value</th>\n",
       "      <th>TotalN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000033</td>\n",
       "      <td>1</td>\n",
       "      <td>226580387</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8266</td>\n",
       "      <td>-0.0574</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.09867</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000050</td>\n",
       "      <td>1</td>\n",
       "      <td>162736463</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.8545</td>\n",
       "      <td>0.0654</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>0.15640</td>\n",
       "      <td>10864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000070</td>\n",
       "      <td>1</td>\n",
       "      <td>222359612</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3360</td>\n",
       "      <td>-0.0229</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.45800</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000073</td>\n",
       "      <td>1</td>\n",
       "      <td>157255396</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3969</td>\n",
       "      <td>-0.0057</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.84730</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000075</td>\n",
       "      <td>1</td>\n",
       "      <td>95166832</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3424</td>\n",
       "      <td>-0.0353</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.20890</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       SNPID  CHR         BP Allele1 Allele2   Freq1  Effect  StdErr  P.value  \\\n",
       "0  rs1000033    1  226580387       t       g  0.8266 -0.0574  0.0348  0.09867   \n",
       "1  rs1000050    1  162736463       t       c  0.8545  0.0654  0.0461  0.15640   \n",
       "2  rs1000070    1  222359612       t       c  0.3360 -0.0229  0.0308  0.45800   \n",
       "3  rs1000073    1  157255396       a       g  0.3969 -0.0057  0.0297  0.84730   \n",
       "4  rs1000075    1   95166832       t       c  0.3424 -0.0353  0.0281  0.20890   \n",
       "\n",
       "    TotalN  \n",
       "0  17310.0  \n",
       "1  10864.0  \n",
       "2  17310.0  \n",
       "3  12819.0  \n",
       "4  17310.0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anx = pd.read_csv('anxiety.meta.full.cc.tbl', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f84c7d5cea62ca15",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T14:48:38.590763800Z",
     "start_time": "2024-01-16T14:48:38.528208Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNPID</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Freq1</th>\n",
       "      <th>Effect</th>\n",
       "      <th>StdErr</th>\n",
       "      <th>P.value</th>\n",
       "      <th>TotalN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000033</td>\n",
       "      <td>1</td>\n",
       "      <td>226580387</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8266</td>\n",
       "      <td>-0.0574</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.09867</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000050</td>\n",
       "      <td>1</td>\n",
       "      <td>162736463</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.8545</td>\n",
       "      <td>0.0654</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>0.15640</td>\n",
       "      <td>10864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000070</td>\n",
       "      <td>1</td>\n",
       "      <td>222359612</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3360</td>\n",
       "      <td>-0.0229</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.45800</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000073</td>\n",
       "      <td>1</td>\n",
       "      <td>157255396</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3969</td>\n",
       "      <td>-0.0057</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.84730</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000075</td>\n",
       "      <td>1</td>\n",
       "      <td>95166832</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3424</td>\n",
       "      <td>-0.0353</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.20890</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs1000085</td>\n",
       "      <td>1</td>\n",
       "      <td>66857915</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.2147</td>\n",
       "      <td>-0.0298</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.39260</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs1000127</td>\n",
       "      <td>1</td>\n",
       "      <td>63432716</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.7161</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.67840</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs1000184</td>\n",
       "      <td>1</td>\n",
       "      <td>8640831</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.2466</td>\n",
       "      <td>0.0712</td>\n",
       "      <td>0.0302</td>\n",
       "      <td>0.01846</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs1000282</td>\n",
       "      <td>1</td>\n",
       "      <td>209894785</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.7900</td>\n",
       "      <td>-0.0194</td>\n",
       "      <td>0.0325</td>\n",
       "      <td>0.55060</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs1000283</td>\n",
       "      <td>1</td>\n",
       "      <td>209894661</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.2124</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>0.0322</td>\n",
       "      <td>0.77070</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>rs1000312</td>\n",
       "      <td>1</td>\n",
       "      <td>205505477</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3493</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.86170</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs1000313</td>\n",
       "      <td>1</td>\n",
       "      <td>15405489</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7940</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0399</td>\n",
       "      <td>0.74030</td>\n",
       "      <td>10864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs1000314</td>\n",
       "      <td>1</td>\n",
       "      <td>15405907</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7909</td>\n",
       "      <td>0.0093</td>\n",
       "      <td>0.0363</td>\n",
       "      <td>0.79750</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs1000315</td>\n",
       "      <td>1</td>\n",
       "      <td>15406111</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.0663</td>\n",
       "      <td>0.0537</td>\n",
       "      <td>0.0664</td>\n",
       "      <td>0.41840</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs1000348</td>\n",
       "      <td>1</td>\n",
       "      <td>15041242</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.6124</td>\n",
       "      <td>-0.0332</td>\n",
       "      <td>0.0271</td>\n",
       "      <td>0.22150</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs1000352</td>\n",
       "      <td>1</td>\n",
       "      <td>209675578</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9045</td>\n",
       "      <td>0.0643</td>\n",
       "      <td>0.0445</td>\n",
       "      <td>0.14840</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs1000360</td>\n",
       "      <td>1</td>\n",
       "      <td>216228343</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.4052</td>\n",
       "      <td>0.0408</td>\n",
       "      <td>0.0316</td>\n",
       "      <td>0.19650</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs1000364</td>\n",
       "      <td>1</td>\n",
       "      <td>55769791</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.6047</td>\n",
       "      <td>-0.0045</td>\n",
       "      <td>0.0270</td>\n",
       "      <td>0.86830</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>rs1000370</td>\n",
       "      <td>1</td>\n",
       "      <td>40998406</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3670</td>\n",
       "      <td>0.0588</td>\n",
       "      <td>0.0300</td>\n",
       "      <td>0.04993</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs1000382</td>\n",
       "      <td>1</td>\n",
       "      <td>175866321</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7423</td>\n",
       "      <td>-0.0530</td>\n",
       "      <td>0.0340</td>\n",
       "      <td>0.11870</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs1000417</td>\n",
       "      <td>1</td>\n",
       "      <td>34007886</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7821</td>\n",
       "      <td>-0.0206</td>\n",
       "      <td>0.0343</td>\n",
       "      <td>0.54910</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs1000418</td>\n",
       "      <td>1</td>\n",
       "      <td>34008297</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7810</td>\n",
       "      <td>-0.0120</td>\n",
       "      <td>0.0316</td>\n",
       "      <td>0.70400</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs1000431</td>\n",
       "      <td>1</td>\n",
       "      <td>28225114</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3827</td>\n",
       "      <td>0.0223</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.41140</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs1000432</td>\n",
       "      <td>1</td>\n",
       "      <td>28225119</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.6193</td>\n",
       "      <td>-0.0243</td>\n",
       "      <td>0.0272</td>\n",
       "      <td>0.37160</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs1000451</td>\n",
       "      <td>1</td>\n",
       "      <td>244187847</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.9458</td>\n",
       "      <td>-0.2521</td>\n",
       "      <td>0.1368</td>\n",
       "      <td>0.06523</td>\n",
       "      <td>2770.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>rs1000452</td>\n",
       "      <td>1</td>\n",
       "      <td>244188049</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.7154</td>\n",
       "      <td>-0.0359</td>\n",
       "      <td>0.0304</td>\n",
       "      <td>0.23770</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs1000476</td>\n",
       "      <td>1</td>\n",
       "      <td>58921516</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9314</td>\n",
       "      <td>-0.0732</td>\n",
       "      <td>0.0626</td>\n",
       "      <td>0.24190</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs1000494</td>\n",
       "      <td>1</td>\n",
       "      <td>29789167</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5664</td>\n",
       "      <td>0.0397</td>\n",
       "      <td>0.0267</td>\n",
       "      <td>0.13760</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs10005</td>\n",
       "      <td>1</td>\n",
       "      <td>230416953</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5782</td>\n",
       "      <td>-0.0245</td>\n",
       "      <td>0.0314</td>\n",
       "      <td>0.43430</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs1000528</td>\n",
       "      <td>1</td>\n",
       "      <td>114471189</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.7002</td>\n",
       "      <td>-0.0570</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.06415</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>rs1000533</td>\n",
       "      <td>1</td>\n",
       "      <td>168282491</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.6208</td>\n",
       "      <td>0.0276</td>\n",
       "      <td>0.0292</td>\n",
       "      <td>0.34340</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs1000543</td>\n",
       "      <td>1</td>\n",
       "      <td>244187600</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5197</td>\n",
       "      <td>0.0254</td>\n",
       "      <td>0.0323</td>\n",
       "      <td>0.43290</td>\n",
       "      <td>10864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs1000606</td>\n",
       "      <td>1</td>\n",
       "      <td>93772193</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.1854</td>\n",
       "      <td>-0.0527</td>\n",
       "      <td>0.0339</td>\n",
       "      <td>0.11990</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>rs1000607</td>\n",
       "      <td>1</td>\n",
       "      <td>93772001</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3363</td>\n",
       "      <td>-0.0113</td>\n",
       "      <td>0.0277</td>\n",
       "      <td>0.68430</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>rs1000730</td>\n",
       "      <td>1</td>\n",
       "      <td>231963601</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3611</td>\n",
       "      <td>-0.0259</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.38290</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>rs1000731</td>\n",
       "      <td>1</td>\n",
       "      <td>231963491</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.2453</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.0329</td>\n",
       "      <td>0.68830</td>\n",
       "      <td>15355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>rs1000792</td>\n",
       "      <td>1</td>\n",
       "      <td>100040100</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8517</td>\n",
       "      <td>-0.0748</td>\n",
       "      <td>0.0409</td>\n",
       "      <td>0.06766</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>rs1000796</td>\n",
       "      <td>1</td>\n",
       "      <td>217402330</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3186</td>\n",
       "      <td>0.0160</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.56910</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>rs1000836</td>\n",
       "      <td>1</td>\n",
       "      <td>237834407</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.2790</td>\n",
       "      <td>-0.0481</td>\n",
       "      <td>0.0585</td>\n",
       "      <td>0.41120</td>\n",
       "      <td>6748.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>rs1000876</td>\n",
       "      <td>1</td>\n",
       "      <td>58719296</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.6586</td>\n",
       "      <td>0.0454</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.13970</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>rs1000895</td>\n",
       "      <td>1</td>\n",
       "      <td>191568187</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.4296</td>\n",
       "      <td>0.0131</td>\n",
       "      <td>0.0264</td>\n",
       "      <td>0.62090</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>rs1000947</td>\n",
       "      <td>1</td>\n",
       "      <td>112277597</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.6533</td>\n",
       "      <td>-0.0347</td>\n",
       "      <td>0.0335</td>\n",
       "      <td>0.29980</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>rs1000997</td>\n",
       "      <td>1</td>\n",
       "      <td>16125961</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.7183</td>\n",
       "      <td>-0.0273</td>\n",
       "      <td>0.0458</td>\n",
       "      <td>0.55180</td>\n",
       "      <td>10663.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>rs1001056</td>\n",
       "      <td>1</td>\n",
       "      <td>18850123</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9420</td>\n",
       "      <td>0.2116</td>\n",
       "      <td>0.1725</td>\n",
       "      <td>0.21980</td>\n",
       "      <td>1955.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>rs1001070</td>\n",
       "      <td>1</td>\n",
       "      <td>78518432</td>\n",
       "      <td>a</td>\n",
       "      <td>t</td>\n",
       "      <td>0.9252</td>\n",
       "      <td>-0.0125</td>\n",
       "      <td>0.0565</td>\n",
       "      <td>0.82520</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>rs1001103</td>\n",
       "      <td>1</td>\n",
       "      <td>78519889</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.0789</td>\n",
       "      <td>0.0148</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.78660</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>rs1001104</td>\n",
       "      <td>1</td>\n",
       "      <td>78519875</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.9214</td>\n",
       "      <td>-0.0125</td>\n",
       "      <td>0.0548</td>\n",
       "      <td>0.81990</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>rs1001134</td>\n",
       "      <td>1</td>\n",
       "      <td>222334294</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.6485</td>\n",
       "      <td>0.0130</td>\n",
       "      <td>0.0283</td>\n",
       "      <td>0.64510</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>rs1001135</td>\n",
       "      <td>1</td>\n",
       "      <td>222334181</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3510</td>\n",
       "      <td>-0.0131</td>\n",
       "      <td>0.0283</td>\n",
       "      <td>0.64350</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>rs1001136</td>\n",
       "      <td>1</td>\n",
       "      <td>222334134</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3481</td>\n",
       "      <td>-0.0199</td>\n",
       "      <td>0.0292</td>\n",
       "      <td>0.49580</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        SNPID  CHR         BP Allele1 Allele2   Freq1  Effect  StdErr  \\\n",
       "0   rs1000033    1  226580387       t       g  0.8266 -0.0574  0.0348   \n",
       "1   rs1000050    1  162736463       t       c  0.8545  0.0654  0.0461   \n",
       "2   rs1000070    1  222359612       t       c  0.3360 -0.0229  0.0308   \n",
       "3   rs1000073    1  157255396       a       g  0.3969 -0.0057  0.0297   \n",
       "4   rs1000075    1   95166832       t       c  0.3424 -0.0353  0.0281   \n",
       "5   rs1000085    1   66857915       c       g  0.2147 -0.0298  0.0348   \n",
       "6   rs1000127    1   63432716       t       c  0.7161  0.0123  0.0297   \n",
       "7   rs1000184    1    8640831       c       g  0.2466  0.0712  0.0302   \n",
       "8   rs1000282    1  209894785       t       c  0.7900 -0.0194  0.0325   \n",
       "9   rs1000283    1  209894661       a       g  0.2124  0.0094  0.0322   \n",
       "10  rs1000312    1  205505477       a       g  0.3493  0.0052  0.0297   \n",
       "11  rs1000313    1   15405489       a       g  0.7940  0.0132  0.0399   \n",
       "12  rs1000314    1   15405907       a       g  0.7909  0.0093  0.0363   \n",
       "13  rs1000315    1   15406111       t       c  0.0663  0.0537  0.0664   \n",
       "14  rs1000348    1   15041242       c       g  0.6124 -0.0332  0.0271   \n",
       "15  rs1000352    1  209675578       t       c  0.9045  0.0643  0.0445   \n",
       "16  rs1000360    1  216228343       c       g  0.4052  0.0408  0.0316   \n",
       "17  rs1000364    1   55769791       t       c  0.6047 -0.0045  0.0270   \n",
       "18  rs1000370    1   40998406       t       g  0.3670  0.0588  0.0300   \n",
       "19  rs1000382    1  175866321       c       g  0.7423 -0.0530  0.0340   \n",
       "20  rs1000417    1   34007886       a       g  0.7821 -0.0206  0.0343   \n",
       "21  rs1000418    1   34008297       a       g  0.7810 -0.0120  0.0316   \n",
       "22  rs1000431    1   28225114       t       c  0.3827  0.0223  0.0272   \n",
       "23  rs1000432    1   28225119       t       c  0.6193 -0.0243  0.0272   \n",
       "24  rs1000451    1  244187847       c       g  0.9458 -0.2521  0.1368   \n",
       "25  rs1000452    1  244188049       t       c  0.7154 -0.0359  0.0304   \n",
       "26  rs1000476    1   58921516       t       c  0.9314 -0.0732  0.0626   \n",
       "27  rs1000494    1   29789167       t       g  0.5664  0.0397  0.0267   \n",
       "28    rs10005    1  230416953       a       g  0.5782 -0.0245  0.0314   \n",
       "29  rs1000528    1  114471189       t       c  0.7002 -0.0570  0.0308   \n",
       "30  rs1000533    1  168282491       t       c  0.6208  0.0276  0.0292   \n",
       "31  rs1000543    1  244187600       a       g  0.5197  0.0254  0.0323   \n",
       "32  rs1000606    1   93772193       t       c  0.1854 -0.0527  0.0339   \n",
       "33  rs1000607    1   93772001       t       c  0.3363 -0.0113  0.0277   \n",
       "34  rs1000730    1  231963601       t       c  0.3611 -0.0259  0.0297   \n",
       "35  rs1000731    1  231963491       t       c  0.2453  0.0132  0.0329   \n",
       "36  rs1000792    1  100040100       a       g  0.8517 -0.0748  0.0409   \n",
       "37  rs1000796    1  217402330       a       g  0.3186  0.0160  0.0281   \n",
       "38  rs1000836    1  237834407       a       g  0.2790 -0.0481  0.0585   \n",
       "39  rs1000876    1   58719296       c       g  0.6586  0.0454  0.0308   \n",
       "40  rs1000895    1  191568187       a       g  0.4296  0.0131  0.0264   \n",
       "41  rs1000947    1  112277597       t       c  0.6533 -0.0347  0.0335   \n",
       "42  rs1000997    1   16125961       a       g  0.7183 -0.0273  0.0458   \n",
       "43  rs1001056    1   18850123       t       c  0.9420  0.2116  0.1725   \n",
       "44  rs1001070    1   78518432       a       t  0.9252 -0.0125  0.0565   \n",
       "45  rs1001103    1   78519889       t       c  0.0789  0.0148  0.0547   \n",
       "46  rs1001104    1   78519875       c       g  0.9214 -0.0125  0.0548   \n",
       "47  rs1001134    1  222334294       a       g  0.6485  0.0130  0.0283   \n",
       "48  rs1001135    1  222334181       t       c  0.3510 -0.0131  0.0283   \n",
       "49  rs1001136    1  222334134       a       g  0.3481 -0.0199  0.0292   \n",
       "\n",
       "    P.value   TotalN  \n",
       "0   0.09867  17310.0  \n",
       "1   0.15640  10864.0  \n",
       "2   0.45800  17310.0  \n",
       "3   0.84730  12819.0  \n",
       "4   0.20890  17310.0  \n",
       "5   0.39260  15355.0  \n",
       "6   0.67840  17310.0  \n",
       "7   0.01846  17310.0  \n",
       "8   0.55060  17310.0  \n",
       "9   0.77070  17310.0  \n",
       "10  0.86170  17310.0  \n",
       "11  0.74030  10864.0  \n",
       "12  0.79750  12819.0  \n",
       "13  0.41840  12819.0  \n",
       "14  0.22150  17310.0  \n",
       "15  0.14840  17310.0  \n",
       "16  0.19650  12819.0  \n",
       "17  0.86830  17310.0  \n",
       "18  0.04993  12819.0  \n",
       "19  0.11870  17310.0  \n",
       "20  0.54910  15355.0  \n",
       "21  0.70400  17310.0  \n",
       "22  0.41140  17310.0  \n",
       "23  0.37160  17310.0  \n",
       "24  0.06523   2770.0  \n",
       "25  0.23770  17310.0  \n",
       "26  0.24190  12819.0  \n",
       "27  0.13760  17310.0  \n",
       "28  0.43430  12819.0  \n",
       "29  0.06415  15355.0  \n",
       "30  0.34340  15355.0  \n",
       "31  0.43290  10864.0  \n",
       "32  0.11990  17310.0  \n",
       "33  0.68430  17310.0  \n",
       "34  0.38290  15355.0  \n",
       "35  0.68830  15355.0  \n",
       "36  0.06766  17310.0  \n",
       "37  0.56910  17310.0  \n",
       "38  0.41120   6748.0  \n",
       "39  0.13970  12819.0  \n",
       "40  0.62090  17310.0  \n",
       "41  0.29980  12819.0  \n",
       "42  0.55180  10663.0  \n",
       "43  0.21980   1955.0  \n",
       "44  0.82520  12819.0  \n",
       "45  0.78660  12819.0  \n",
       "46  0.81990  12819.0  \n",
       "47  0.64510  17310.0  \n",
       "48  0.64350  17310.0  \n",
       "49  0.49580  17310.0  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anx.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "29af25d3c7d3a98e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T14:45:19.702985800Z",
     "start_time": "2024-01-16T14:44:45.927249Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anx.to_csv('anxiety.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9cf7742ca78060c6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anxiety_df = pd.read_table('Summary_statistics/raw/anxiety.meta.full.cc.tbl')\n",
    "anxiety_df = anxiety_df[['SNPID', 'P.value']]\n",
    "anxiety_df = anxiety_df.rename(columns={'SNPID':'rs_id', 'P.value':'p_value'})\n",
    "psych_dfs.append(anxiety_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5921e6411ec0acfd",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "aud_df = pd.read_csv('Summary_statistics/raw/AUDIT_UKB_2018_AJP.txt', sep=' ')\n",
    "aud_df = aud_df[['rsid', 'p_T']]\n",
    "aud_df = aud_df.rename(columns={'rsid':'rs_id', 'p_T':'p_value'})\n",
    "psych_dfs.append(aud_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7de573d05514d5f7",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anorexia_df = pd.read_csv('Summary_statistics/raw/pgcAN2.2019-07.vcf.tsv', sep='\\t', comment='#')\n",
    "anorexia_df = anorexia_df[['ID', 'PVAL']]\n",
    "anorexia_df = anorexia_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(anorexia_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9add3a15205f54d",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "asd_df = pd.read_csv('Summary_statistics/raw/iPSYCH-PGC_ASD_Nov2017', sep='\\t')\n",
    "asd_df = asd_df[['SNP', 'P']]\n",
    "asd_df = asd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(asd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1e0758079fa65a5",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "bpd_df = pd.read_csv('Summary_statistics/raw/daner_bip_pgc3_nm_noukbiobank', sep='\\t')\n",
    "bpd_df = bpd_df[['SNP', 'P']]\n",
    "bpd_df = bpd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(bpd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3313b4b92ee1d0ab",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cud_df = pd.read_csv('Summary_statistics/raw/CUD_EUR_full_public_11.14.2020', sep=' ')\n",
    "cud_df = cud_df[['SNP', 'P']]\n",
    "cud_df = cud_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(cud_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c69da02fe8fa698",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "hoarding_df = pd.read_csv('Summary_statistics/raw/hoarding2022.vcf.tsv', sep='\\t', comment='#')\n",
    "hoarding_df = hoarding_df[['ID', 'PVAL']]\n",
    "hoarding_df = hoarding_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(hoarding_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa437654985d641c",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mdd_df = pd.read_csv('Summary_statistics/raw/PGC_UKB_depression_genome-wide.txt', sep=' ')\n",
    "mdd_df = mdd_df[['MarkerName', 'P']]\n",
    "mdd_df = mdd_df.rename(columns={'MarkerName':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(mdd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfffe2dd7fb48b0b",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ocd_df = pd.read_csv('Summary_statistics/raw/ocd_aug2017', sep='\\t')\n",
    "ocd_df = ocd_df[['SNP', 'P']]\n",
    "ocd_df = ocd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(ocd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b68538017cc99c96",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:39:47.401528400Z",
     "start_time": "2024-01-23T08:39:25.521541400Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "od_1 = pd.read_table('opi.DEPvEXP_EUR.noAF.tbl')\n",
    "od_2 = pd.read_table('opi.DEPvUNX_EUR.noAF.tbl')\n",
    "od_3 = pd.read_table('opi.EXPvUNX_EUR.noAF.tbl')\n",
    "# od_1 = od_1[['rsID', 'P-value']]\n",
    "# od_2 = od_2[['rsID', 'P-value']]\n",
    "# od_3 = od_3[['rsID', 'P-value']]\n",
    "\n",
    "od_list =[od_1, od_2, od_3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f2df2e2e70e120a5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:42:32.093377800Z",
     "start_time": "2024-01-23T08:42:30.109524700Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "od_df = pd.concat(od_list)\n",
    "# od_df = od_df[['rsID', 'P-value']]\n",
    "# od_df = od_df.rename(columns={'rsID':'rs_id', 'P-value':'p_value'})\n",
    "# psych_dfs.append(od_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "81fa127ce72dbfc5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:46:19.669306200Z",
     "start_time": "2024-01-23T08:44:40.179371900Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "od_df.to_csv('opium_dependence.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adbc976ce4dbca1b",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ptsd_df = pd.read_csv('Summary_statistics/raw/pts_eur_freeze2_overall.results', sep='\\t')\n",
    "ptsd_df = ptsd_df[['SNP', 'P']]\n",
    "ptsd_df = ptsd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(ptsd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c494075ff79cb1a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scz_df = pd.read_csv('Summary_statistics/raw/PGC3_SCZ_wave3.european.autosome.public.v3.vcf.tsv', sep='\\t', comment='#')\n",
    "scz_df = scz_df[['ID', 'PVAL']]\n",
    "scz_df = scz_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(scz_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71c0345af0b1fc0d",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tourette_df = pd.read_csv('Summary_statistics/raw/TS_Oct2018', sep=' ')\n",
    "tourette_df = tourette_df[['SNP', 'P']]\n",
    "tourette_df = tourette_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(tourette_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c2b32f452fcafa0",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "panic_df = pd.read_csv('Summary_statistics/raw/pgc-panic2019.vcf.tsv', sep='\\t', header=72)\n",
    "panic_df = panic_df[['ID', 'PVAL']]\n",
    "panic_df = panic_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(panic_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "341696174d6f12b0",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#lists of disorder names\n",
    "psych_names = ['adhd', 'asd', 'anorexia', 'anxiety', 'aud', 'bpd', 'cud', 'hoarding', 'mdd', 'ocd', 'od', 'ptsd', 'scz', 'tourette']\n",
    "other_names = ['epilepsy','sjogrens', 'lupus', 'biliary_chorrhosis', 'myasthenia', 'rh_arthritis', 'ms', 'parkinsons','migraine', 'fibromyalgia', 'fatigue', 'ibs', 'gout']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22ed364930e8689a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Add a column with the name of the disorder in each dataframe\n",
    "for df, name in zip(psych_dfs, psych_names):\n",
    "    df['disorder'] = name\n",
    "\n",
    "# for df, name in zip(other_dfs, other_names):\n",
    "#     df['disorder'] = name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7b2f22dbccf024a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psych_dfs[9].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a91b83aee35a942",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df = pd. read_csv('psych_pivoted.csv',index_col='rs_id',nrows=200)\n",
    "df = df.fillna(0)\n",
    "df = df.drop(columns='Unnamed: 0')\n",
    "df.head()\n",
    "\n",
    "# correlation_matrix = df.corr(method='spearman')\n",
    "#\n",
    "# plt.figure(figsize=(15, 15))\n",
    "#\n",
    "#\n",
    "# sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', fmt=\".2f\")\n",
    "#\n",
    "#\n",
    "# plt.title('Correlation Heatmap')\n",
    "#\n",
    "#\n",
    "# plt.savefig('correlation_heatmap.jpg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd3362330cf66733",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
