{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4151d2bcebe5c238",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T07:50:21.546396900Z",
     "start_time": "2024-01-23T07:50:21.546396900Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f83ae74642b3d5e",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psych_dfs = []\n",
    "other_dfs = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21d4ee7399a8c6fc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-15T08:20:32.780644400Z",
     "start_time": "2024-01-15T08:20:17.540341700Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "# fibromyalgia = pd.read_csv('Summary_statistics/raw/fibromyalgia.tsv', sep='\\t', comment='#') #need to map rsID\n",
    "# fatigue = pd.read_csv('Summary_statistics/raw/chronic_fatigue.tsv', sep='\\t', comment='#') #need to map rsID\n",
    "# gout = pd.read_csv('Summary_statistics/raw/gout.tsv', sep='\\t') #need to map rsID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6e87d7a674a67e6c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:37:01.921092700Z",
     "start_time": "2024-01-22T12:36:46.601890Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sjogrens_df= pd.read_csv('sjogrens_buildGRCh37.tsv', sep='\\t')\n",
    "#sjogrens_df = sjogrens_df[['variant_id', 'p_value']]\n",
    "sjogrens_df = sjogrens_df.rename(columns={'chromosome':'chromosome','variant_id':'variant_id','base_pair_location':'base_pair_location', 'other_allele':'other_allele', 'effect_allele':'effect_allele', 'effect_allele_frequency':'effect_allele_frequency', 'INFO':'INFO', 'beta':'beta', 'standard_error':'standard_error', 'T':'T', '-log10p':'-log10p', 'p_value':'p_value'})\n",
    "# other_dfs.append(sjogrens_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c7ce350e63a567ab",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:38:55.511717200Z",
     "start_time": "2024-01-22T12:37:45.724507100Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sjogrens_df.to_csv('sjogrens.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "fc54da49c8034dc2",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T12:37:05.784112800Z",
     "start_time": "2024-01-22T12:37:05.768309600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 9419702 entries, 0 to 9419701\n",
      "Data columns (total 12 columns):\n",
      " #   Column                   Dtype  \n",
      "---  ------                   -----  \n",
      " 0   chromosome               int64  \n",
      " 1   variant_id               object \n",
      " 2   base_pair_location       int64  \n",
      " 3   other_allele             object \n",
      " 4   effect_allele            object \n",
      " 5   effect_allele_frequency  float64\n",
      " 6   INFO                     float64\n",
      " 7   beta                     float64\n",
      " 8   standard_error           float64\n",
      " 9   T                        float64\n",
      " 10  -log10p                  float64\n",
      " 11  p_value                  float64\n",
      "dtypes: float64(7), int64(2), object(3)\n",
      "memory usage: 862.4+ MB\n"
     ]
    }
   ],
   "source": [
    "sjogrens_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a184843d98147bae",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>rs_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>130840091</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-1.321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8493</td>\n",
       "      <td>0.1866</td>\n",
       "      <td>rs2326918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7</td>\n",
       "      <td>145771806</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-1.294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8541</td>\n",
       "      <td>0.1956</td>\n",
       "      <td>rs6977693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>100009976</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8720</td>\n",
       "      <td>0.8532</td>\n",
       "      <td>rs12364336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>166367755</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>1.379</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7572</td>\n",
       "      <td>0.1680</td>\n",
       "      <td>rs12562373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "      <td>86737556</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1772</td>\n",
       "      <td>0.9208</td>\n",
       "      <td>rs2135099</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location effect_allele other_allele   beta  \\\n",
       "0           6           130840091             a            g -1.321   \n",
       "1           7           145771806             t            c -1.294   \n",
       "2          11           100009976             a            g  0.185   \n",
       "3           1           166367755             a            g  1.379   \n",
       "4          14            86737556             a            g  0.099   \n",
       "\n",
       "   standard_error  effect_allele_frequency  p_value       rs_id  \n",
       "0             NaN                   0.8493   0.1866   rs2326918  \n",
       "1             NaN                   0.8541   0.1956   rs6977693  \n",
       "2             NaN                   0.8720   0.8532  rs12364336  \n",
       "3             NaN                   0.7572   0.1680  rs12562373  \n",
       "4             NaN                   0.1772   0.9208   rs2135099  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "epilepsy_df = pd.read_csv('epilepsy.tsv', sep='\\t', nrows=5)\n",
    "#epilepsy_df = epilepsy_df[['rs_id', 'p_value']]\n",
    "#other_dfs.append(epilepsy_df)\n",
    "epilepsy_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1b65773c5af44995",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1lele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs12562034</td>\n",
       "      <td>768448</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.915</td>\n",
       "      <td>0.3604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs7518545</td>\n",
       "      <td>769963</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.904</td>\n",
       "      <td>0.3660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs12124819</td>\n",
       "      <td>776546</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>-0.075</td>\n",
       "      <td>0.9405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs117086422</td>\n",
       "      <td>845635</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-0.777</td>\n",
       "      <td>0.4374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs28612348</td>\n",
       "      <td>846078</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>-0.826</td>\n",
       "      <td>0.4090</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR          SNP      BP A1lele1 Allele2  Zscore  P-value\n",
       "0    1   rs12562034  768448       a       g  -0.915   0.3604\n",
       "1    1    rs7518545  769963       a       g  -0.904   0.3660\n",
       "2    1   rs12124819  776546       a       g  -0.075   0.9405\n",
       "3    1  rs117086422  845635       t       c  -0.777   0.4374\n",
       "4    1   rs28612348  846078       t       c  -0.826   0.4090"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lupus_df = pd.read_csv('systemic_lupus.txt', sep='\\t', nrows=5)\n",
    "\"\"\" lupus_df = lupus_df[['SNP', 'P-value']]\n",
    "lupus_df = lupus_df.rename(columns={'SNP':'rs_id','P-value':'p_value'})\n",
    "other_dfs.append(lupus_df) \"\"\"\n",
    "lupus_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a5a356aa146bc32a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 5054572 entries, 0 to 5054571\n",
      "Data columns (total 8 columns):\n",
      " #   Column              Dtype  \n",
      "---  ------              -----  \n",
      " 0   variant_id          object \n",
      " 1   chromosome          int64  \n",
      " 2   base_pair_location  int64  \n",
      " 3   effect_allele       object \n",
      " 4   other_allele        object \n",
      " 5   p_value             float64\n",
      " 6   beta                float64\n",
      " 7   standard_error      float64\n",
      "dtypes: float64(3), int64(2), object(3)\n",
      "memory usage: 308.5+ MB\n"
     ]
    }
   ],
   "source": [
    "prim_bil_df = pd.read_csv('biliary_chirrosis.tsv', sep='\\t', comment='#')\n",
    "\"\"\" prim_bil_df = prim_bil_df[['variant_id', 'p_value']]\n",
    "prim_bil_df = prim_bil_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(prim_bil_df) \"\"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2119948e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_id            0\n",
       "chromosome            0\n",
       "base_pair_location    0\n",
       "effect_allele         0\n",
       "other_allele          0\n",
       "p_value               0\n",
       "beta                  0\n",
       "standard_error        0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prim_bil_df.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "2f6ca9eff20f88c1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-17T09:28:00.232235Z",
     "start_time": "2024-01-17T09:27:37.096483600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_id                 518903\n",
       "chromosome                      0\n",
       "base_pair_location              0\n",
       "GENPOS                          0\n",
       "effect_allele                   0\n",
       "other_allele                    0\n",
       "effect_allele_frequency         0\n",
       "INFO                            0\n",
       "CHISQ_LINREG                    0\n",
       "P_LINREG                        0\n",
       "beta                            0\n",
       "standard_error                  0\n",
       "CHISQ_BOLT_LMM_INF              0\n",
       "P_BOLT_LMM_INF                  0\n",
       "CHISQ_BOLT_LMM                  0\n",
       "p_value                         0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thyroid_df = pd.read_csv('thyroid_buildGRCh37.tsv', sep='\\t', comment='#')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d56bd563",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "variant_id                 0\n",
       "chromosome                 0\n",
       "base_pair_location         0\n",
       "GENPOS                     0\n",
       "effect_allele              0\n",
       "other_allele               0\n",
       "effect_allele_frequency    0\n",
       "INFO                       0\n",
       "CHISQ_LINREG               0\n",
       "P_LINREG                   0\n",
       "beta                       0\n",
       "standard_error             0\n",
       "CHISQ_BOLT_LMM_INF         0\n",
       "P_BOLT_LMM_INF             0\n",
       "CHISQ_BOLT_LMM             0\n",
       "p_value                    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thyroid_df = thyroid_df.loc[thyroid_df['p_value'] <= 0.05]\n",
    "thyroid_df = thyroid_df[thyroid_df['variant_id'].notna()]\n",
    "thyroid_df.reset_index(drop=True, inplace=True)\n",
    "# thyroid_df = thyroid_df[['variant_id', 'p_value']]\n",
    "# thyroid_df = thyroid_df.rename(columns={'variant_id':'rs_id'})\n",
    "# other_dfs.append(thyroid_df)\n",
    "thyroid_df.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "07fb4930",
   "metadata": {},
   "outputs": [],
   "source": [
    "thyroid_df.to_csv('thyroid_filtered_and_notna.csv', sep=' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f008872c9ee88173",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-22T13:01:48.451685300Z",
     "start_time": "2024-01-22T13:01:47.169691800Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scleroderma_df = pd.read_csv('systemic_scleroderma.txt', header=0, delim_whitespace=True)\n",
    "scleroderma_df = scleroderma_df[['SNP', 'P', 'A1', 'A2', 'OR']]\n",
    "# scleroderma_df = scleroderma_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "# other_dfs.append(scleroderma_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a544155ade05b6f6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>effect_allele_frequency_cases</th>\n",
       "      <th>effect_allele_frequency_controls</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs61769340</td>\n",
       "      <td>1</td>\n",
       "      <td>727717</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.132900</td>\n",
       "      <td>0.144688</td>\n",
       "      <td>0.132293</td>\n",
       "      <td>0.1063</td>\n",
       "      <td>0.0522</td>\n",
       "      <td>0.041610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>rs116030099</td>\n",
       "      <td>1</td>\n",
       "      <td>787290</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.079988</td>\n",
       "      <td>0.063801</td>\n",
       "      <td>0.080822</td>\n",
       "      <td>-0.1757</td>\n",
       "      <td>0.0736</td>\n",
       "      <td>0.017020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>rs148120343</td>\n",
       "      <td>1</td>\n",
       "      <td>794707</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.044400</td>\n",
       "      <td>0.036839</td>\n",
       "      <td>0.044790</td>\n",
       "      <td>0.1894</td>\n",
       "      <td>0.0966</td>\n",
       "      <td>0.049970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>rs79010578</td>\n",
       "      <td>1</td>\n",
       "      <td>800909</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.095547</td>\n",
       "      <td>0.084357</td>\n",
       "      <td>0.096123</td>\n",
       "      <td>0.1344</td>\n",
       "      <td>0.0653</td>\n",
       "      <td>0.039550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>rs61768170</td>\n",
       "      <td>1</td>\n",
       "      <td>821054</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.105143</td>\n",
       "      <td>0.102776</td>\n",
       "      <td>0.105265</td>\n",
       "      <td>0.1569</td>\n",
       "      <td>0.0602</td>\n",
       "      <td>0.009184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "9     rs61769340           1              727717             G            C   \n",
       "52   rs116030099           1              787290             C            T   \n",
       "59   rs148120343           1              794707             C            T   \n",
       "76    rs79010578           1              800909             A            T   \n",
       "117   rs61768170           1              821054             C            G   \n",
       "\n",
       "     effect_allele_frequency  effect_allele_frequency_cases  \\\n",
       "9                   0.132900                       0.144688   \n",
       "52                  0.079988                       0.063801   \n",
       "59                  0.044400                       0.036839   \n",
       "76                  0.095547                       0.084357   \n",
       "117                 0.105143                       0.102776   \n",
       "\n",
       "     effect_allele_frequency_controls    beta  standard_error   p_value  \n",
       "9                            0.132293  0.1063          0.0522  0.041610  \n",
       "52                           0.080822 -0.1757          0.0736  0.017020  \n",
       "59                           0.044790  0.1894          0.0966  0.049970  \n",
       "76                           0.096123  0.1344          0.0653  0.039550  \n",
       "117                          0.105265  0.1569          0.0602  0.009184  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myasthenia_df = pd.read_csv('myastheria_gravis.tsv', sep='\\t', comment='#')\n",
    "myasthenia_df = myasthenia_df[myasthenia_df['p_value'] <= 0.05]\n",
    "\"\"\" myasthenia_df = myasthenia_df[['variant_id', 'p_value']]\n",
    "myasthenia_df = myasthenia_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(myasthenia_df) \"\"\"\n",
    "myasthenia_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9c830a6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "myasthenia_df.to_csv('myasthenia_gravis_filtered.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ebde0194b6646a7a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id_hg19</th>\n",
       "      <th>base_pair_location_grch38</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs144804129</td>\n",
       "      <td>10</td>\n",
       "      <td>100000122</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.2710</td>\n",
       "      <td>0.4979</td>\n",
       "      <td>0.5863</td>\n",
       "      <td>10_100000122_T_A</td>\n",
       "      <td>98240365.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs6602381</td>\n",
       "      <td>10</td>\n",
       "      <td>10000018</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0008</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>0.9550</td>\n",
       "      <td>10_10000018_A_G</td>\n",
       "      <td>9958055.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs11442554</td>\n",
       "      <td>10</td>\n",
       "      <td>100000554</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.0217</td>\n",
       "      <td>0.1945</td>\n",
       "      <td>10_100000554_A_AT</td>\n",
       "      <td>98240797.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs112832083</td>\n",
       "      <td>10</td>\n",
       "      <td>100000588</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.5309</td>\n",
       "      <td>0.9955</td>\n",
       "      <td>0.5938</td>\n",
       "      <td>10_100000588_T_C</td>\n",
       "      <td>98240831.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7899632</td>\n",
       "      <td>10</td>\n",
       "      <td>100000625</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0070</td>\n",
       "      <td>0.0153</td>\n",
       "      <td>0.6492</td>\n",
       "      <td>10_100000625_A_G</td>\n",
       "      <td>98240868.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0  rs144804129          10           100000122             A            T   \n",
       "1    rs6602381          10            10000018             G            A   \n",
       "2   rs11442554          10           100000554            AT            A   \n",
       "3  rs112832083          10           100000588             C            T   \n",
       "4    rs7899632          10           100000625             G            A   \n",
       "\n",
       "     beta  standard_error  p_value    variant_id_hg19  \\\n",
       "0 -0.2710          0.4979   0.5863   10_100000122_T_A   \n",
       "1 -0.0008          0.0143   0.9550    10_10000018_A_G   \n",
       "2  0.0281          0.0217   0.1945  10_100000554_A_AT   \n",
       "3 -0.5309          0.9955   0.5938   10_100000588_T_C   \n",
       "4 -0.0070          0.0153   0.6492   10_100000625_A_G   \n",
       "\n",
       "   base_pair_location_grch38  \n",
       "0                 98240365.0  \n",
       "1                  9958055.0  \n",
       "2                 98240797.0  \n",
       "3                 98240831.0  \n",
       "4                 98240868.0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rh_arthritis_df = pd.read_csv('rheumatoid_arthritis.tsv', sep='\\t', comment='#')\n",
    "\"\"\" rh_arthritis_df = rh_arthritis_df[['variant_id', 'p_value']]\n",
    "rh_arthritis_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(rh_arthritis_df) \"\"\"\n",
    "rh_arthritis_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "f5483e0a41f87001",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\" ms_df = ms_df[['variant_id', 'p_value']]\\nms_df = ms_df.rename(columns={'variant_id':'rs_id'})\\nother_dfs.append(ms_df) \""
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ms_df= pd.read_csv('ms.tsv', sep='\\t', comment='#')\n",
    "ms_df = ms_df[ms_df['variant_id'].notna()]\n",
    "ms_df.reset_index(drop=True, inplace=True)\n",
    "\"\"\" ms_df = ms_df[['variant_id', 'p_value']]\n",
    "ms_df = ms_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(ms_df) \"\"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "5fdd0224",
   "metadata": {},
   "outputs": [],
   "source": [
    "ms_df.to_csv('ms_not_null.txt', sep=' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "34d8e130067df282",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T07:51:15.689525200Z",
     "start_time": "2024-01-23T07:51:12.007797300Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>SNP_HRC_ID</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>LOG(OR)_SE</th>\n",
       "      <th>ci_lower</th>\n",
       "      <th>ci_upper</th>\n",
       "      <th>Z_STAT</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>762320</td>\n",
       "      <td>1:762320:C:T</td>\n",
       "      <td>rs75333668</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.15582</td>\n",
       "      <td>0.341344</td>\n",
       "      <td>0.592022</td>\n",
       "      <td>2.25653</td>\n",
       "      <td>0.424232</td>\n",
       "      <td>0.671396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>765028</td>\n",
       "      <td>1:765028:C:T</td>\n",
       "      <td>rs2519025</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.28589</td>\n",
       "      <td>0.418656</td>\n",
       "      <td>0.566036</td>\n",
       "      <td>2.92123</td>\n",
       "      <td>0.600618</td>\n",
       "      <td>0.548094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>780229</td>\n",
       "      <td>1:780229:T:C</td>\n",
       "      <td>rs12565032</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.30387</td>\n",
       "      <td>0.419182</td>\n",
       "      <td>0.573356</td>\n",
       "      <td>2.96511</td>\n",
       "      <td>0.632978</td>\n",
       "      <td>0.526748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>782677</td>\n",
       "      <td>1:782677:G:T</td>\n",
       "      <td>rs140041012</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.21011</td>\n",
       "      <td>0.407450</td>\n",
       "      <td>0.544505</td>\n",
       "      <td>2.68934</td>\n",
       "      <td>0.468054</td>\n",
       "      <td>0.639746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865628</td>\n",
       "      <td>1:865628:G:A</td>\n",
       "      <td>rs41285790</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10533</td>\n",
       "      <td>1.15970</td>\n",
       "      <td>0.165675</td>\n",
       "      <td>0.838150</td>\n",
       "      <td>1.60460</td>\n",
       "      <td>0.894276</td>\n",
       "      <td>0.371174</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location    SNP_HRC_ID   variant_id REF ALT  \\\n",
       "0           1              762320  1:762320:C:T   rs75333668   C   T   \n",
       "1           1              765028  1:765028:C:T    rs2519025   C   T   \n",
       "2           1              780229  1:780229:T:C   rs12565032   T   C   \n",
       "3           1              782677  1:782677:G:T  rs140041012   G   T   \n",
       "4           1              865628  1:865628:G:A   rs41285790   G   A   \n",
       "\n",
       "  effect_allele TEST  OBS_CT  odds_ratio  LOG(OR)_SE  ci_lower  ci_upper  \\\n",
       "0             T  ADD   10533     1.15582    0.341344  0.592022   2.25653   \n",
       "1             T  ADD   10533     1.28589    0.418656  0.566036   2.92123   \n",
       "2             C  ADD   10533     1.30387    0.419182  0.573356   2.96511   \n",
       "3             T  ADD   10533     1.21011    0.407450  0.544505   2.68934   \n",
       "4             A  ADD   10533     1.15970    0.165675  0.838150   1.60460   \n",
       "\n",
       "     Z_STAT   p_value  \n",
       "0  0.424232  0.671396  \n",
       "1  0.600618  0.548094  \n",
       "2  0.632978  0.526748  \n",
       "3  0.468054  0.639746  \n",
       "4  0.894276  0.371174  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parkinsons_df = pd.read_csv('parkinson.tsv',sep='\\t', comment='#', nrows=5)\n",
    "# parkinsons_df = parkinsons_df[['variant_id', 'p_value']]\n",
    "# parkinsons_df.rename(columns={'variant_id':'rs_id'})\n",
    "# other_dfs.append(parkinsons_df)\n",
    "parkinsons_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b1611444a91dc1c8",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>rs201400700</td>\n",
       "      <td>1</td>\n",
       "      <td>732134</td>\n",
       "      <td>0.004426</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.985016</td>\n",
       "      <td>0.550936</td>\n",
       "      <td>6.29879</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>-0.004629</td>\n",
       "      <td>0.001884</td>\n",
       "      <td>6.03976</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>6.07759</td>\n",
       "      <td>0.0140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>rs369030935</td>\n",
       "      <td>1</td>\n",
       "      <td>732989</td>\n",
       "      <td>0.004452</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.975057</td>\n",
       "      <td>0.663507</td>\n",
       "      <td>3.80110</td>\n",
       "      <td>0.0510</td>\n",
       "      <td>-0.002679</td>\n",
       "      <td>0.001337</td>\n",
       "      <td>4.01755</td>\n",
       "      <td>0.0450</td>\n",
       "      <td>4.06075</td>\n",
       "      <td>0.0440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>rs2427928</td>\n",
       "      <td>1</td>\n",
       "      <td>735068</td>\n",
       "      <td>0.004516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.979575</td>\n",
       "      <td>0.523948</td>\n",
       "      <td>4.04231</td>\n",
       "      <td>0.0440</td>\n",
       "      <td>0.003478</td>\n",
       "      <td>0.001701</td>\n",
       "      <td>4.18371</td>\n",
       "      <td>0.0410</td>\n",
       "      <td>4.33075</td>\n",
       "      <td>0.0370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>rs181876450</td>\n",
       "      <td>1</td>\n",
       "      <td>736689</td>\n",
       "      <td>0.004566</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.989025</td>\n",
       "      <td>0.702724</td>\n",
       "      <td>8.96721</td>\n",
       "      <td>0.0027</td>\n",
       "      <td>0.005821</td>\n",
       "      <td>0.001943</td>\n",
       "      <td>8.97823</td>\n",
       "      <td>0.0027</td>\n",
       "      <td>9.03015</td>\n",
       "      <td>0.0027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>rs3964475</td>\n",
       "      <td>1</td>\n",
       "      <td>743021</td>\n",
       "      <td>0.004733</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.771204</td>\n",
       "      <td>0.512939</td>\n",
       "      <td>4.82817</td>\n",
       "      <td>0.0280</td>\n",
       "      <td>-0.001231</td>\n",
       "      <td>0.000568</td>\n",
       "      <td>4.70095</td>\n",
       "      <td>0.0300</td>\n",
       "      <td>4.71240</td>\n",
       "      <td>0.0300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     variant_id  chromosome  base_pair_location    GENPOS effect_allele  \\\n",
       "74  rs201400700           1              732134  0.004426             T   \n",
       "77  rs369030935           1              732989  0.004452             C   \n",
       "80    rs2427928           1              735068  0.004516             A   \n",
       "83  rs181876450           1              736689  0.004566             T   \n",
       "93    rs3964475           1              743021  0.004733             T   \n",
       "\n",
       "   other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "74           TA                 0.985016  0.550936       6.29879    0.0120   \n",
       "77            T                 0.975057  0.663507       3.80110    0.0510   \n",
       "80            G                 0.979575  0.523948       4.04231    0.0440   \n",
       "83            C                 0.989025  0.702724       8.96721    0.0027   \n",
       "93            C                 0.771204  0.512939       4.82817    0.0280   \n",
       "\n",
       "        beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "74 -0.004629        0.001884             6.03976          0.0140   \n",
       "77 -0.002679        0.001337             4.01755          0.0450   \n",
       "80  0.003478        0.001701             4.18371          0.0410   \n",
       "83  0.005821        0.001943             8.97823          0.0027   \n",
       "93 -0.001231        0.000568             4.70095          0.0300   \n",
       "\n",
       "    CHISQ_BOLT_LMM  p_value  \n",
       "74         6.07759   0.0140  \n",
       "77         4.06075   0.0440  \n",
       "80         4.33075   0.0370  \n",
       "83         9.03015   0.0027  \n",
       "93         4.71240   0.0300  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "migraine_df = pd.read_csv('migraine.tsv',sep='\\t', comment='#')\n",
    "migraine_df = migraine_df[migraine_df['p_value'] <= 0.05]\n",
    "\"\"\" migraine_df = migraine_df[['variant_id', 'p_value']]\n",
    "migraine_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(migraine_df) \"\"\"\n",
    "migraine_df.to_csv('migraine_filtered_pval.txt', sep='\\t')\n",
    "migraine_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4d271fe5b78d60a7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T09:42:03.767827600Z",
     "start_time": "2024-01-16T09:41:57.646604100Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>134897743</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.1725</td>\n",
       "      <td>-0.1574</td>\n",
       "      <td>0.1952</td>\n",
       "      <td>0.4199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>8600861</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.4160</td>\n",
       "      <td>0.1298</td>\n",
       "      <td>0.1511</td>\n",
       "      <td>0.3904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>100009976</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8734</td>\n",
       "      <td>0.3009</td>\n",
       "      <td>0.2185</td>\n",
       "      <td>0.1685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>3840048</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5911</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.1487</td>\n",
       "      <td>0.9941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>20776538</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.2802</td>\n",
       "      <td>-0.0839</td>\n",
       "      <td>0.1670</td>\n",
       "      <td>0.6155</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location effect_allele other_allele  \\\n",
       "0          11           134897743             c            g   \n",
       "1          16             8600861             a            c   \n",
       "2          11           100009976             a            g   \n",
       "3          12             3840048             a            g   \n",
       "4           7            20776538             a            c   \n",
       "\n",
       "   effect_allele_frequency    beta  standard_error  p_value  \n",
       "0                   0.1725 -0.1574          0.1952   0.4199  \n",
       "1                   0.4160  0.1298          0.1511   0.3904  \n",
       "2                   0.8734  0.3009          0.2185   0.1685  \n",
       "3                   0.5911  0.0011          0.1487   0.9941  \n",
       "4                   0.2802 -0.0839          0.1670   0.6155  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fibromyalgia_df = pd.read_csv('fibromyalgia_buildGRCh37.tsv', sep='\\t')\n",
    "fibromyalgia_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "8a6184f81415d85b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T10:47:39.964581400Z",
     "start_time": "2024-01-16T10:47:18.407440500Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 487390 entries, 0 to 487389\n",
      "Data columns (total 14 columns):\n",
      " #   Column                   Non-Null Count   Dtype  \n",
      "---  ------                   --------------   -----  \n",
      " 0   variant_id               487390 non-null  object \n",
      " 1   GENPOS                   487390 non-null  float64\n",
      " 2   effect_allele            487390 non-null  object \n",
      " 3   other_allele             487390 non-null  object \n",
      " 4   effect_allele_frequency  487390 non-null  float64\n",
      " 5   INFO                     487390 non-null  float64\n",
      " 6   CHISQ_LINREG             487390 non-null  float64\n",
      " 7   P_LINREG                 487390 non-null  float64\n",
      " 8   beta                     487390 non-null  float64\n",
      " 9   standard_error           487390 non-null  float64\n",
      " 10  CHISQ_BOLT_LMM_INF       487390 non-null  float64\n",
      " 11  P_BOLT_LMM_INF           487390 non-null  float64\n",
      " 12  CHISQ_BOLT_LMM           487390 non-null  float64\n",
      " 13  p_value                  487390 non-null  float64\n",
      "dtypes: float64(11), object(3)\n",
      "memory usage: 52.1+ MB\n"
     ]
    }
   ],
   "source": [
    "fatigue_df = pd.read_csv('fatigue_buildGRCh37.tsv', sep='\\t')\n",
    "fatigue_df = fatigue_df.drop(columns=['chromosome', 'base_pair_location'])\n",
    "fatigue_df = fatigue_df.loc[fatigue_df['p_value'] <= 0.05]\n",
    "fatigue_df = fatigue_df[fatigue_df['variant_id'].notna()]\n",
    "fatigue_df.reset_index(drop=True, inplace=True)\n",
    "\"\"\" fatigue_df = fatigue_df[['variant_id', 'p_value']]\n",
    "fatigue_df = fatigue_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(fatigue_df)\n",
    " \"\"\"\n",
    "fatigue_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "cd0a6611",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs189259765</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.723683</td>\n",
       "      <td>0.508780</td>\n",
       "      <td>4.99315</td>\n",
       "      <td>0.0250</td>\n",
       "      <td>0.000475</td>\n",
       "      <td>0.000212</td>\n",
       "      <td>5.01303</td>\n",
       "      <td>0.0250</td>\n",
       "      <td>5.03554</td>\n",
       "      <td>0.0250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs200129875</td>\n",
       "      <td>0.004906</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.592892</td>\n",
       "      <td>0.525374</td>\n",
       "      <td>6.90706</td>\n",
       "      <td>0.0086</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>0.000188</td>\n",
       "      <td>6.78899</td>\n",
       "      <td>0.0092</td>\n",
       "      <td>6.75710</td>\n",
       "      <td>0.0093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs57181708</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.895195</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.43869</td>\n",
       "      <td>0.0350</td>\n",
       "      <td>0.000461</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>4.42652</td>\n",
       "      <td>0.0350</td>\n",
       "      <td>4.48090</td>\n",
       "      <td>0.0340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs72888853</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.984519</td>\n",
       "      <td>0.558540</td>\n",
       "      <td>7.15932</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>-0.001951</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>7.21962</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>7.21795</td>\n",
       "      <td>0.0072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs535862693</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.984519</td>\n",
       "      <td>0.558540</td>\n",
       "      <td>7.15932</td>\n",
       "      <td>0.0075</td>\n",
       "      <td>-0.001951</td>\n",
       "      <td>0.000726</td>\n",
       "      <td>7.21962</td>\n",
       "      <td>0.0072</td>\n",
       "      <td>7.21795</td>\n",
       "      <td>0.0072</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id    GENPOS effect_allele other_allele  effect_allele_frequency  \\\n",
       "0  rs189259765  0.000000             C            T                 0.723683   \n",
       "1  rs200129875  0.004906             A            C                 0.592892   \n",
       "2   rs57181708  0.006208             A            G                 0.895195   \n",
       "3   rs72888853  0.006208             T            C                 0.984519   \n",
       "4  rs535862693  0.006208             A            G                 0.984519   \n",
       "\n",
       "       INFO  CHISQ_LINREG  P_LINREG      beta  standard_error  \\\n",
       "0  0.508780       4.99315    0.0250  0.000475        0.000212   \n",
       "1  0.525374       6.90706    0.0086  0.000491        0.000188   \n",
       "2  1.000000       4.43869    0.0350  0.000461        0.000219   \n",
       "3  0.558540       7.15932    0.0075 -0.001951        0.000726   \n",
       "4  0.558540       7.15932    0.0075 -0.001951        0.000726   \n",
       "\n",
       "   CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  CHISQ_BOLT_LMM  p_value  \n",
       "0             5.01303          0.0250         5.03554   0.0250  \n",
       "1             6.78899          0.0092         6.75710   0.0093  \n",
       "2             4.42652          0.0350         4.48090   0.0340  \n",
       "3             7.21962          0.0072         7.21795   0.0072  \n",
       "4             7.21962          0.0072         7.21795   0.0072  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fatigue_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "d365b3e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "fatigue_df.to_csv('fatigue_without_chrom_base_pair_loc.txt', sep=' ', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2ed39d47fdf783ba",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>p_value</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>N_CASE</th>\n",
       "      <th>N_CONTROL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs367896724</td>\n",
       "      <td>0.1537</td>\n",
       "      <td>1</td>\n",
       "      <td>10177</td>\n",
       "      <td>A</td>\n",
       "      <td>AC</td>\n",
       "      <td>0.6024</td>\n",
       "      <td>0.0171</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>40548</td>\n",
       "      <td>293220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs201106462</td>\n",
       "      <td>0.5196</td>\n",
       "      <td>1</td>\n",
       "      <td>10352</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.6076</td>\n",
       "      <td>0.0079</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>40548</td>\n",
       "      <td>293220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs575272151</td>\n",
       "      <td>0.7053</td>\n",
       "      <td>1</td>\n",
       "      <td>11008</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9137</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0.0205</td>\n",
       "      <td>40548</td>\n",
       "      <td>293220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs544419019</td>\n",
       "      <td>0.7053</td>\n",
       "      <td>1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9137</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0.0205</td>\n",
       "      <td>40548</td>\n",
       "      <td>293220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs540538026</td>\n",
       "      <td>0.3802</td>\n",
       "      <td>1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0590</td>\n",
       "      <td>0.0238</td>\n",
       "      <td>0.0271</td>\n",
       "      <td>40548</td>\n",
       "      <td>293220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  p_value  chromosome  base_pair_location effect_allele  \\\n",
       "0  rs367896724   0.1537           1               10177             A   \n",
       "1  rs201106462   0.5196           1               10352             T   \n",
       "2  rs575272151   0.7053           1               11008             C   \n",
       "3  rs544419019   0.7053           1               11012             C   \n",
       "4  rs540538026   0.3802           1               13110             A   \n",
       "\n",
       "  other_allele  effect_allele_frequency    beta  standard_error  N_CASE  \\\n",
       "0           AC                   0.6024  0.0171          0.0120   40548   \n",
       "1           TA                   0.6076  0.0079          0.0123   40548   \n",
       "2            G                   0.9137  0.0077          0.0205   40548   \n",
       "3            G                   0.9137  0.0077          0.0205   40548   \n",
       "4            G                   0.0590  0.0238          0.0271   40548   \n",
       "\n",
       "   N_CONTROL  \n",
       "0     293220  \n",
       "1     293220  \n",
       "2     293220  \n",
       "3     293220  \n",
       "4     293220  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ibs_df = pd.read_csv('IBS.tsv', sep='\\t', comment='#')\n",
    "ibs_df.dropna()\n",
    "\"\"\" ibs_df = ibs_df[['variant_id', 'p_value']]\n",
    "ibs_df = ibs_df.rename(columns={'variant_id':'rs_id'})\n",
    "other_dfs.append(ibs_df) \"\"\"\n",
    "ibs_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e530c6b2ca0853eb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-15T11:06:23.465586100Z",
     "start_time": "2024-01-15T11:06:02.056037600Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>INFO</th>\n",
       "      <th>CHISQ_LINREG</th>\n",
       "      <th>P_LINREG</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>CHISQ_BOLT_LMM_INF</th>\n",
       "      <th>P_BOLT_LMM_INF</th>\n",
       "      <th>CHISQ_BOLT_LMM</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs533316052</td>\n",
       "      <td>1</td>\n",
       "      <td>834430</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.981954</td>\n",
       "      <td>0.902227</td>\n",
       "      <td>4.88949</td>\n",
       "      <td>0.0270</td>\n",
       "      <td>-0.002142</td>\n",
       "      <td>0.000940</td>\n",
       "      <td>5.19438</td>\n",
       "      <td>0.0230</td>\n",
       "      <td>4.99975</td>\n",
       "      <td>0.0250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs565071104</td>\n",
       "      <td>1</td>\n",
       "      <td>845509</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.840534</td>\n",
       "      <td>0.738628</td>\n",
       "      <td>5.25631</td>\n",
       "      <td>0.0220</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>5.19605</td>\n",
       "      <td>0.0230</td>\n",
       "      <td>5.75915</td>\n",
       "      <td>0.0160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs145881258</td>\n",
       "      <td>1</td>\n",
       "      <td>876067</td>\n",
       "      <td>0.006208</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.928461</td>\n",
       "      <td>0.844886</td>\n",
       "      <td>6.86200</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>0.001325</td>\n",
       "      <td>0.000503</td>\n",
       "      <td>6.94770</td>\n",
       "      <td>0.0084</td>\n",
       "      <td>6.74929</td>\n",
       "      <td>0.0094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>885882</td>\n",
       "      <td>0.006287</td>\n",
       "      <td>GAC</td>\n",
       "      <td>G</td>\n",
       "      <td>0.974464</td>\n",
       "      <td>0.887075</td>\n",
       "      <td>6.18209</td>\n",
       "      <td>0.0130</td>\n",
       "      <td>0.001964</td>\n",
       "      <td>0.000801</td>\n",
       "      <td>6.00773</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>5.85623</td>\n",
       "      <td>0.0160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs148363371</td>\n",
       "      <td>1</td>\n",
       "      <td>889876</td>\n",
       "      <td>0.006370</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.989011</td>\n",
       "      <td>0.955070</td>\n",
       "      <td>3.69856</td>\n",
       "      <td>0.0540</td>\n",
       "      <td>0.002260</td>\n",
       "      <td>0.001166</td>\n",
       "      <td>3.75791</td>\n",
       "      <td>0.0530</td>\n",
       "      <td>3.84921</td>\n",
       "      <td>0.0500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id  chromosome  base_pair_location    GENPOS effect_allele  \\\n",
       "0  rs533316052           1              834430  0.006208             G   \n",
       "1  rs565071104           1              845509  0.006208             T   \n",
       "2  rs145881258           1              876067  0.006208             C   \n",
       "3          NaN           1              885882  0.006287           GAC   \n",
       "4  rs148363371           1              889876  0.006370             C   \n",
       "\n",
       "  other_allele  effect_allele_frequency      INFO  CHISQ_LINREG  P_LINREG  \\\n",
       "0            A                 0.981954  0.902227       4.88949    0.0270   \n",
       "1            G                 0.840534  0.738628       5.25631    0.0220   \n",
       "2            T                 0.928461  0.844886       6.86200    0.0088   \n",
       "3            G                 0.974464  0.887075       6.18209    0.0130   \n",
       "4            G                 0.989011  0.955070       3.69856    0.0540   \n",
       "\n",
       "       beta  standard_error  CHISQ_BOLT_LMM_INF  P_BOLT_LMM_INF  \\\n",
       "0 -0.002142        0.000940             5.19438          0.0230   \n",
       "1  0.000861        0.000378             5.19605          0.0230   \n",
       "2  0.001325        0.000503             6.94770          0.0084   \n",
       "3  0.001964        0.000801             6.00773          0.0140   \n",
       "4  0.002260        0.001166             3.75791          0.0530   \n",
       "\n",
       "   CHISQ_BOLT_LMM  p_value  \n",
       "0         4.99975   0.0250  \n",
       "1         5.75915   0.0160  \n",
       "2         6.74929   0.0094  \n",
       "3         5.85623   0.0160  \n",
       "4         3.84921   0.0500  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gout_df = pd.read_csv('gout_buildGRCh37.tsv', sep='\\t')\n",
    "gout_df = gout_df.loc[gout_df['p_value'] <= 0.05]\n",
    "gout_df.reset_index(inplace=True, drop= True)\n",
    "gout_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f7124516",
   "metadata": {},
   "outputs": [],
   "source": [
    "gout_df.to_csv('gout_filtered_by_pval.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e89bc4c6b43060a6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Psychiatric disorder GWAS dataset\n",
    "\n",
    "# alz = pd.read_csv('Summary_statistics/raw/PGCALZ2sumstatsExcluding23andMe.txt', sep='\\t') #need to map rsID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e04651a8e7dd7e18",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "adhd_df = pd.read_csv('Summary_statistics/raw/ADHD_meta_Jan2022_iPSYCH1_iPSYCH2_deCODE_PGC.meta', sep=' ')\n",
    "adhd_df = adhd_df[['SNP', 'P']]\n",
    "adhd_df = adhd_df.rename(columns={'SNP': 'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(adhd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b1105d2174a37f15",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T14:41:24.956773200Z",
     "start_time": "2024-01-16T14:41:16.865298100Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNPID</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Freq1</th>\n",
       "      <th>Effect</th>\n",
       "      <th>StdErr</th>\n",
       "      <th>P.value</th>\n",
       "      <th>TotalN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000033</td>\n",
       "      <td>1</td>\n",
       "      <td>226580387</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.8266</td>\n",
       "      <td>-0.0574</td>\n",
       "      <td>0.0348</td>\n",
       "      <td>0.09867</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1000050</td>\n",
       "      <td>1</td>\n",
       "      <td>162736463</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.8545</td>\n",
       "      <td>0.0654</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>0.15640</td>\n",
       "      <td>10864.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000070</td>\n",
       "      <td>1</td>\n",
       "      <td>222359612</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3360</td>\n",
       "      <td>-0.0229</td>\n",
       "      <td>0.0308</td>\n",
       "      <td>0.45800</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000073</td>\n",
       "      <td>1</td>\n",
       "      <td>157255396</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.3969</td>\n",
       "      <td>-0.0057</td>\n",
       "      <td>0.0297</td>\n",
       "      <td>0.84730</td>\n",
       "      <td>12819.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000075</td>\n",
       "      <td>1</td>\n",
       "      <td>95166832</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.3424</td>\n",
       "      <td>-0.0353</td>\n",
       "      <td>0.0281</td>\n",
       "      <td>0.20890</td>\n",
       "      <td>17310.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       SNPID  CHR         BP Allele1 Allele2   Freq1  Effect  StdErr  P.value  \\\n",
       "0  rs1000033    1  226580387       t       g  0.8266 -0.0574  0.0348  0.09867   \n",
       "1  rs1000050    1  162736463       t       c  0.8545  0.0654  0.0461  0.15640   \n",
       "2  rs1000070    1  222359612       t       c  0.3360 -0.0229  0.0308  0.45800   \n",
       "3  rs1000073    1  157255396       a       g  0.3969 -0.0057  0.0297  0.84730   \n",
       "4  rs1000075    1   95166832       t       c  0.3424 -0.0353  0.0281  0.20890   \n",
       "\n",
       "    TotalN  \n",
       "0  17310.0  \n",
       "1  10864.0  \n",
       "2  17310.0  \n",
       "3  12819.0  \n",
       "4  17310.0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anx = pd.read_csv('anxiety.meta.full.cc.tbl', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "29af25d3c7d3a98e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-16T14:45:19.702985800Z",
     "start_time": "2024-01-16T14:44:45.927249Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anx.to_csv('anxiety.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9cf7742ca78060c6",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anxiety_df = pd.read_table('Summary_statistics/raw/anxiety.meta.full.cc.tbl')\n",
    "anxiety_df = anxiety_df[['SNPID', 'P.value']]\n",
    "anxiety_df = anxiety_df.rename(columns={'SNPID':'rs_id', 'P.value':'p_value'})\n",
    "psych_dfs.append(anxiety_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5921e6411ec0acfd",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "aud_df = pd.read_csv('Summary_statistics/raw/AUDIT_UKB_2018_AJP.txt', sep=' ')\n",
    "aud_df = aud_df[['rsid', 'p_T']]\n",
    "aud_df = aud_df.rename(columns={'rsid':'rs_id', 'p_T':'p_value'})\n",
    "psych_dfs.append(aud_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7de573d05514d5f7",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "anorexia_df = pd.read_csv('Summary_statistics/raw/pgcAN2.2019-07.vcf.tsv', sep='\\t', comment='#')\n",
    "anorexia_df = anorexia_df[['ID', 'PVAL']]\n",
    "anorexia_df = anorexia_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(anorexia_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9add3a15205f54d",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "asd_df = pd.read_csv('Summary_statistics/raw/iPSYCH-PGC_ASD_Nov2017', sep='\\t')\n",
    "asd_df = asd_df[['SNP', 'P']]\n",
    "asd_df = asd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(asd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1e0758079fa65a5",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "bpd_df = pd.read_csv('Summary_statistics/raw/daner_bip_pgc3_nm_noukbiobank', sep='\\t')\n",
    "bpd_df = bpd_df[['SNP', 'P']]\n",
    "bpd_df = bpd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(bpd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3313b4b92ee1d0ab",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cud_df = pd.read_csv('Summary_statistics/raw/CUD_EUR_full_public_11.14.2020', sep=' ')\n",
    "cud_df = cud_df[['SNP', 'P']]\n",
    "cud_df = cud_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(cud_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c69da02fe8fa698",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "hoarding_df = pd.read_csv('Summary_statistics/raw/hoarding2022.vcf.tsv', sep='\\t', comment='#')\n",
    "hoarding_df = hoarding_df[['ID', 'PVAL']]\n",
    "hoarding_df = hoarding_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(hoarding_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa437654985d641c",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mdd_df = pd.read_csv('Summary_statistics/raw/PGC_UKB_depression_genome-wide.txt', sep=' ')\n",
    "mdd_df = mdd_df[['MarkerName', 'P']]\n",
    "mdd_df = mdd_df.rename(columns={'MarkerName':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(mdd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfffe2dd7fb48b0b",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ocd_df = pd.read_csv('Summary_statistics/raw/ocd_aug2017', sep='\\t')\n",
    "ocd_df = ocd_df[['SNP', 'P']]\n",
    "ocd_df = ocd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(ocd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b68538017cc99c96",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:39:47.401528400Z",
     "start_time": "2024-01-23T08:39:25.521541400Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "od_df = pd.read_table('opi.DEPvUNX_EUR.noAF.tbl')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1cff826f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Weight</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "      <th>HetISq</th>\n",
       "      <th>HetChiSq</th>\n",
       "      <th>HetDf</th>\n",
       "      <th>HetPVal</th>\n",
       "      <th>Total_N</th>\n",
       "      <th>Total_NCase</th>\n",
       "      <th>Total_NControl</th>\n",
       "      <th>ngt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "      <td>4.887055e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.433896e+03</td>\n",
       "      <td>1.225321e-02</td>\n",
       "      <td>4.912236e-01</td>\n",
       "      <td>2.165467e+00</td>\n",
       "      <td>5.807185e+00</td>\n",
       "      <td>5.729918e+00</td>\n",
       "      <td>4.953439e-01</td>\n",
       "      <td>1.141511e+04</td>\n",
       "      <td>2.659696e+03</td>\n",
       "      <td>8.755415e+03</td>\n",
       "      <td>3.373216e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.741449e+02</td>\n",
       "      <td>1.031735e+00</td>\n",
       "      <td>2.915502e-01</td>\n",
       "      <td>1.691245e+01</td>\n",
       "      <td>3.724466e+00</td>\n",
       "      <td>1.359486e+00</td>\n",
       "      <td>2.898796e-01</td>\n",
       "      <td>1.462363e+03</td>\n",
       "      <td>5.785281e+01</td>\n",
       "      <td>1.424176e+03</td>\n",
       "      <td>8.842286e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>3.807080e+03</td>\n",
       "      <td>-5.180000e+00</td>\n",
       "      <td>1.615000e-07</td>\n",
       "      <td>-6.670000e+01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>1.710000e-09</td>\n",
       "      <td>6.332000e+03</td>\n",
       "      <td>2.101000e+03</td>\n",
       "      <td>3.836000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>4.311930e+03</td>\n",
       "      <td>-6.810000e-01</td>\n",
       "      <td>2.367000e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.049000e+00</td>\n",
       "      <td>5.000000e+00</td>\n",
       "      <td>2.434000e-01</td>\n",
       "      <td>1.008400e+04</td>\n",
       "      <td>2.633000e+03</td>\n",
       "      <td>7.456000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>4.484910e+03</td>\n",
       "      <td>1.100000e-02</td>\n",
       "      <td>4.884000e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>5.114000e+00</td>\n",
       "      <td>6.000000e+00</td>\n",
       "      <td>4.936000e-01</td>\n",
       "      <td>1.114900e+04</td>\n",
       "      <td>2.676000e+03</td>\n",
       "      <td>8.460000e+03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>4.566220e+03</td>\n",
       "      <td>7.050000e-01</td>\n",
       "      <td>7.441000e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7.801000e+00</td>\n",
       "      <td>7.000000e+00</td>\n",
       "      <td>7.467000e-01</td>\n",
       "      <td>1.299800e+04</td>\n",
       "      <td>2.697000e+03</td>\n",
       "      <td>1.030200e+04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>4.613910e+03</td>\n",
       "      <td>5.239000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>8.720000e+01</td>\n",
       "      <td>4.676100e+01</td>\n",
       "      <td>7.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.325200e+04</td>\n",
       "      <td>2.712000e+03</td>\n",
       "      <td>1.054000e+04</td>\n",
       "      <td>5.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Weight        Zscore       P-value        HetISq      HetChiSq  \\\n",
       "count  4.887055e+06  4.887055e+06  4.887055e+06  4.887055e+06  4.887055e+06   \n",
       "mean   4.433896e+03  1.225321e-02  4.912236e-01  2.165467e+00  5.807185e+00   \n",
       "std    1.741449e+02  1.031735e+00  2.915502e-01  1.691245e+01  3.724466e+00   \n",
       "min    3.807080e+03 -5.180000e+00  1.615000e-07 -6.670000e+01  0.000000e+00   \n",
       "25%    4.311930e+03 -6.810000e-01  2.367000e-01  0.000000e+00  3.049000e+00   \n",
       "50%    4.484910e+03  1.100000e-02  4.884000e-01  0.000000e+00  5.114000e+00   \n",
       "75%    4.566220e+03  7.050000e-01  7.441000e-01  0.000000e+00  7.801000e+00   \n",
       "max    4.613910e+03  5.239000e+00  1.000000e+00  8.720000e+01  4.676100e+01   \n",
       "\n",
       "              HetDf       HetPVal       Total_N   Total_NCase  Total_NControl  \\\n",
       "count  4.887055e+06  4.887055e+06  4.887055e+06  4.887055e+06    4.887055e+06   \n",
       "mean   5.729918e+00  4.953439e-01  1.141511e+04  2.659696e+03    8.755415e+03   \n",
       "std    1.359486e+00  2.898796e-01  1.462363e+03  5.785281e+01    1.424176e+03   \n",
       "min    3.000000e+00  1.710000e-09  6.332000e+03  2.101000e+03    3.836000e+03   \n",
       "25%    5.000000e+00  2.434000e-01  1.008400e+04  2.633000e+03    7.456000e+03   \n",
       "50%    6.000000e+00  4.936000e-01  1.114900e+04  2.676000e+03    8.460000e+03   \n",
       "75%    7.000000e+00  7.467000e-01  1.299800e+04  2.697000e+03    1.030200e+04   \n",
       "max    7.000000e+00  1.000000e+00  1.325200e+04  2.712000e+03    1.054000e+04   \n",
       "\n",
       "                ngt  \n",
       "count  4.887055e+06  \n",
       "mean   3.373216e-01  \n",
       "std    8.842286e-01  \n",
       "min    0.000000e+00  \n",
       "25%    0.000000e+00  \n",
       "50%    0.000000e+00  \n",
       "75%    0.000000e+00  \n",
       "max    5.000000e+00  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "od_df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f2df2e2e70e120a5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:42:32.093377800Z",
     "start_time": "2024-01-23T08:42:30.109524700Z"
    },
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13252"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "od_df['Total_N'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "81fa127ce72dbfc5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-01-23T08:46:19.669306200Z",
     "start_time": "2024-01-23T08:44:40.179371900Z"
    },
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "od_df.to_csv('opium_dependence.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "adbc976ce4dbca1b",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'pts_eur_freeze2_overall.results'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[10], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m ptsd_df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mpts_eur_freeze2_overall.results\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\t\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m      2\u001b[0m ptsd_df \u001b[38;5;241m=\u001b[39m ptsd_df[[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSNP\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mP\u001b[39m\u001b[38;5;124m'\u001b[39m]]\n\u001b[0;32m      3\u001b[0m ptsd_df \u001b[38;5;241m=\u001b[39m ptsd_df\u001b[38;5;241m.\u001b[39mrename(columns\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSNP\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mrs_id\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mP\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mp_value\u001b[39m\u001b[38;5;124m'\u001b[39m})\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m    936\u001b[0m     dialect,\n\u001b[0;32m    937\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    944\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[0;32m    945\u001b[0m )\n\u001b[0;32m    946\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m--> 948\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:611\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    608\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[0;32m    610\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[1;32m--> 611\u001b[0m parser \u001b[38;5;241m=\u001b[39m TextFileReader(filepath_or_buffer, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwds)\n\u001b[0;32m    613\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[0;32m    614\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1448\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[1;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[0;32m   1445\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m   1447\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m-> 1448\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1705\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[1;34m(self, f, engine)\u001b[0m\n\u001b[0;32m   1703\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[0;32m   1704\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m-> 1705\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   1706\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1707\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1708\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1709\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1710\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1711\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1712\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1713\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1714\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1715\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1716\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\common.py:863\u001b[0m, in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    858\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m    859\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[0;32m    860\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[0;32m    861\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[0;32m    862\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[1;32m--> 863\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[0;32m    864\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    865\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    866\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    867\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    868\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    869\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    870\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    871\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[0;32m    872\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'pts_eur_freeze2_overall.results'"
     ]
    }
   ],
   "source": [
    "ptsd_df = pd.read_csv('pts_eur_freeze2_overall.results', sep='\\t')\n",
    "ptsd_df = ptsd_df[['SNP', 'P']]\n",
    "ptsd_df = ptsd_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(ptsd_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3c494075ff79cb1a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'PGC3_SCZ_wave3.european.autosome.public.v3.vcf.tsv'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[11], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m scz_df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPGC3_SCZ_wave3.european.autosome.public.v3.vcf.tsv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\t\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcomment\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m#\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;124;03m\"\"\" scz_df = scz_df[['ID', 'PVAL']]\u001b[39;00m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;124;03mscz_df = scz_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\u001b[39;00m\n\u001b[0;32m      4\u001b[0m \u001b[38;5;124;03mpsych_dfs.append(scz_df) \"\"\"\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m    936\u001b[0m     dialect,\n\u001b[0;32m    937\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    944\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[0;32m    945\u001b[0m )\n\u001b[0;32m    946\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m--> 948\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:611\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    608\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[0;32m    610\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[1;32m--> 611\u001b[0m parser \u001b[38;5;241m=\u001b[39m TextFileReader(filepath_or_buffer, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwds)\n\u001b[0;32m    613\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[0;32m    614\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1448\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[1;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[0;32m   1445\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m   1447\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m-> 1448\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1705\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[1;34m(self, f, engine)\u001b[0m\n\u001b[0;32m   1703\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[0;32m   1704\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m-> 1705\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   1706\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1707\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1708\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1709\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1710\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1711\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1712\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1713\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1714\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1715\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1716\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[1;32mc:\\Users\\annei\\anaconda3\\envs\\psych_overlap\\lib\\site-packages\\pandas\\io\\common.py:863\u001b[0m, in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    858\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m    859\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[0;32m    860\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[0;32m    861\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[0;32m    862\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[1;32m--> 863\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[0;32m    864\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    865\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    866\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    867\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    868\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    869\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    870\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    871\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[0;32m    872\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'PGC3_SCZ_wave3.european.autosome.public.v3.vcf.tsv'"
     ]
    }
   ],
   "source": [
    "scz_df = pd.read_csv('PGC3_SCZ_wave3.european.autosome.public.v3.vcf.tsv', sep='\\t', comment='#')\n",
    "\"\"\" scz_df = scz_df[['ID', 'PVAL']]\n",
    "scz_df = scz_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(scz_df) \"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc20221c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71c0345af0b1fc0d",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "tourette_df = pd.read_csv('Summary_statistics/raw/TS_Oct2018', sep=' ')\n",
    "tourette_df = tourette_df[['SNP', 'P']]\n",
    "tourette_df = tourette_df.rename(columns={'SNP':'rs_id', 'P':'p_value'})\n",
    "psych_dfs.append(tourette_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c2b32f452fcafa0",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "panic_df = pd.read_csv('Summary_statistics/raw/pgc-panic2019.vcf.tsv', sep='\\t', header=72)\n",
    "panic_df = panic_df[['ID', 'PVAL']]\n",
    "panic_df = panic_df.rename(columns={'ID':'rs_id', 'PVAL':'p_value'})\n",
    "psych_dfs.append(panic_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "341696174d6f12b0",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#lists of disorder names\n",
    "psych_names = ['adhd', 'asd', 'anorexia', 'anxiety', 'aud', 'bpd', 'cud', 'hoarding', 'mdd', 'ocd', 'od', 'ptsd', 'scz', 'tourette']\n",
    "other_names = ['epilepsy','sjogrens', 'lupus', 'biliary_chorrhosis', 'myasthenia', 'rh_arthritis', 'ms', 'parkinsons','migraine', 'fibromyalgia', 'fatigue', 'ibs', 'gout']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22ed364930e8689a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Add a column with the name of the disorder in each dataframe\n",
    "for df, name in zip(psych_dfs, psych_names):\n",
    "    df['disorder'] = name\n",
    "\n",
    "# for df, name in zip(other_dfs, other_names):\n",
    "#     df['disorder'] = name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7b2f22dbccf024a",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psych_dfs[9].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a91b83aee35a942",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df = pd. read_csv('psych_pivoted.csv',index_col='rs_id',nrows=200)\n",
    "df = df.fillna(0)\n",
    "df = df.drop(columns='Unnamed: 0')\n",
    "df.head()\n",
    "\n",
    "# correlation_matrix = df.corr(method='spearman')\n",
    "#\n",
    "# plt.figure(figsize=(15, 15))\n",
    "#\n",
    "#\n",
    "# sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', fmt=\".2f\")\n",
    "#\n",
    "#\n",
    "# plt.title('Correlation Heatmap')\n",
    "#\n",
    "#\n",
    "# plt.savefig('correlation_heatmap.jpg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd3362330cf66733",
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
